{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.23.4'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['0th loop:ABL1',\n",
       " '0th loop:ALKBH1',\n",
       " '0th loop:ALKBH2',\n",
       " '0th loop:ALKBH3',\n",
       " '0th loop:ACTR5',\n",
       " '0th loop:ACD',\n",
       " '0th loop:ACTL6A',\n",
       " '0th loop:ACTR8',\n",
       " ('ALKBH2', 'subsystem0', 'ddot'),\n",
       " ('ALKBH3', 'subsystem0', 'ddot'),\n",
       " ('ACTR5', 'subsystem1', 'ddot'),\n",
       " ('ACTR8', 'subsystem1', 'ddot'),\n",
       " ('0th posterior',\n",
       "  -1115.2374,\n",
       "  array([[1.6580787 , 3.091736  , 5.157053  , 4.600947  , 2.3965085 ,\n",
       "          3.2050626 ],\n",
       "         [2.7627575 , 0.71248114, 3.6450367 , 3.3966918 , 3.7553399 ,\n",
       "          3.382165  ],\n",
       "         [5.307899  , 3.4954834 , 2.0394535 , 4.594358  , 4.4031787 ,\n",
       "          3.5953705 ],\n",
       "         [3.9513657 , 3.9308922 , 5.107185  , 1.522825  , 5.145914  ,\n",
       "          2.3028893 ],\n",
       "         [2.4188197 , 3.004545  , 3.3142724 , 6.0301223 , 3.2614625 ,\n",
       "          2.837462  ],\n",
       "         [3.922175  , 3.398773  , 3.125641  , 3.1751044 , 2.0222588 ,\n",
       "          4.0695605 ]], dtype=float32),\n",
       "  array([0, 1, 5, 5, 4, 2, 3, 4])),\n",
       " ('ACTL6A', 'subsystem2', 'ddot'),\n",
       " ('ACTR8', 'subsystem2', 'ddot'),\n",
       " ('1th posterior',\n",
       "  -1142.1625,\n",
       "  array([[1.4139422, 4.9560595, 3.473384 , 3.5514197, 3.787266 , 3.0147352],\n",
       "         [4.9534655, 1.2256612, 2.1618152, 2.6276445, 3.7760382, 3.2216282],\n",
       "         [4.4569635, 2.3046675, 1.6009282, 3.1437826, 3.7769568, 3.9673963],\n",
       "         [2.6139486, 2.3748794, 2.9158027, 3.0925772, 4.6281624, 3.2593577],\n",
       "         [3.8076813, 3.2452483, 4.325032 , 4.3335238, 1.5123968, 3.2040143],\n",
       "         [3.2891543, 3.0203214, 4.003042 , 3.3818824, 3.0572226, 4.0728693]],\n",
       "        dtype=float32),\n",
       "  array([0, 1, 5, 5, 4, 2, 3, 3])),\n",
       " '1th loop:ABL1',\n",
       " '1th loop:ALKBH1',\n",
       " '1th loop:ACD',\n",
       " '1th loop:subsystem0',\n",
       " '1th loop:subsystem1',\n",
       " '1th loop:subsystem2',\n",
       " ('ABL1', 'subsystem3', 'ddot'),\n",
       " ('ACD', 'subsystem3', 'ddot'),\n",
       " ('ALKBH1', 'subsystem4', 'ddot'),\n",
       " ('subsystem2', 'subsystem4', 'ddot'),\n",
       " ('subsystem0', 'subsystem5', 'ddot'),\n",
       " ('subsystem1', 'subsystem5', 'ddot'),\n",
       " ('0th posterior',\n",
       "  -363.8266,\n",
       "  array([[1.9961519, 1.6833313, 2.8091886],\n",
       "         [1.272584 , 2.2902145, 1.9002727],\n",
       "         [2.8588576, 2.1325746, 2.3547344]], dtype=float32),\n",
       "  array([0, 1, 0, 2, 2, 1])),\n",
       " '2th loop:subsystem3',\n",
       " '2th loop:subsystem4',\n",
       " '2th loop:subsystem5',\n",
       " ('subsystem3', 'subsystem6', 'ddot'),\n",
       " ('subsystem4', 'subsystem6', 'ddot'),\n",
       " ('subsystem5', 'subsystem6', 'ddot'),\n",
       " ('0th posterior',\n",
       "  -52.080505,\n",
       "  array([[1.7032759]], dtype=float32),\n",
       "  array([0, 0, 0]))]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pickle\n",
    "\n",
    "with open ('resultNew/hierarchy-Bayesian4.save', 'rb') as fp:\n",
    "    itemlist = pickle.load(fp)\n",
    "\n",
    "itemlist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1108"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(mapping)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13 genes, 12 terms, 16 gene-term relations, 6 term-term relations\n",
      "node_attributes: []\n",
      "edge_attributes: []\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>subsystem4</td>\n",
       "      <td>subsystem2</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>subsystem5</td>\n",
       "      <td>subsystem0</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>subsystem5</td>\n",
       "      <td>subsystem1</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>subsystem6</td>\n",
       "      <td>subsystem3</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>subsystem6</td>\n",
       "      <td>subsystem4</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>subsystem6</td>\n",
       "      <td>subsystem5</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>t</td>\n",
       "      <td>0</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>-52.080505</td>\n",
       "      <td>0th posterior</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>-363.8266</td>\n",
       "      <td>0th posterior</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>-1115.2374</td>\n",
       "      <td>0th posterior</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>t</td>\n",
       "      <td>1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>-1142.1625</td>\n",
       "      <td>1th posterior</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>t</td>\n",
       "      <td>2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>subsystem3</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>subsystem3</td>\n",
       "      <td>ACD</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>subsystem2</td>\n",
       "      <td>ACTL6A</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>subsystem1</td>\n",
       "      <td>ACTR5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>subsystem1</td>\n",
       "      <td>ACTR8</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>subsystem2</td>\n",
       "      <td>ACTR8</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>subsystem4</td>\n",
       "      <td>ALKBH1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>subsystem0</td>\n",
       "      <td>ALKBH2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>subsystem0</td>\n",
       "      <td>ALKBH3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Parent          Child      EdgeType\n",
       "0   subsystem4     subsystem2  Child-Parent\n",
       "1   subsystem5     subsystem0  Child-Parent\n",
       "2   subsystem5     subsystem1  Child-Parent\n",
       "3   subsystem6     subsystem3  Child-Parent\n",
       "4   subsystem6     subsystem4  Child-Parent\n",
       "5   subsystem6     subsystem5  Child-Parent\n",
       "6            t              0     Gene-Term\n",
       "7   -52.080505  0th posterior     Gene-Term\n",
       "8    -363.8266  0th posterior     Gene-Term\n",
       "9   -1115.2374  0th posterior     Gene-Term\n",
       "10           t              1     Gene-Term\n",
       "11  -1142.1625  1th posterior     Gene-Term\n",
       "12           t              2     Gene-Term\n",
       "13  subsystem3           ABL1     Gene-Term\n",
       "14  subsystem3            ACD     Gene-Term\n",
       "15  subsystem2         ACTL6A     Gene-Term\n",
       "16  subsystem1          ACTR5     Gene-Term\n",
       "17  subsystem1          ACTR8     Gene-Term\n",
       "18  subsystem2          ACTR8     Gene-Term\n",
       "19  subsystem4         ALKBH1     Gene-Term\n",
       "20  subsystem0         ALKBH2     Gene-Term\n",
       "21  subsystem0         ALKBH3     Gene-Term"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hierarchy  = [x for x in itemlist if \"subsystem\" in str(x[0])]\n",
    "mapping = [x for x in itemlist if \"subsystem\" not in str(x[0])]\n",
    "\n",
    "import ddot\n",
    "from ddot import Ontology\n",
    "# Construct ontology\n",
    "ont = Ontology(hierarchy, mapping)\n",
    "\n",
    "# Prints a summary of the ontology's structure\n",
    "print(ont)\n",
    "ont.to_table('toy_ontology2.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "306 genes, 52 terms, 335 gene-term relations, 92 term-term relations\n",
      "node_attributes: []\n",
      "edge_attributes: []\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>subsystem111</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>subsystem171</td>\n",
       "      <td>subsystem364</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>subsystem180</td>\n",
       "      <td>subsystem111</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>subsystem180</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>subsystem193</td>\n",
       "      <td>subsystem171</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>subsystem207</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>subsystem230</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>subsystem232</td>\n",
       "      <td>subsystem187</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>subsystem232</td>\n",
       "      <td>subsystem201</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>subsystem232</td>\n",
       "      <td>subsystem283</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>subsystem232</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>subsystem283</td>\n",
       "      <td>subsystem180</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>subsystem283</td>\n",
       "      <td>subsystem230</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>subsystem283</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>subsystem284</td>\n",
       "      <td>subsystem235</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>subsystem284</td>\n",
       "      <td>subsystem240</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>subsystem284</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>subsystem319</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>subsystem333</td>\n",
       "      <td>subsystem289</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>subsystem333</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>subsystem334</td>\n",
       "      <td>subsystem193</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>subsystem334</td>\n",
       "      <td>subsystem284</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>subsystem334</td>\n",
       "      <td>subsystem341</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>subsystem13</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>subsystem17</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>subsystem29</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>subsystem51</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>subsystem53</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>subsystem58</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>subsystem63</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>397</th>\n",
       "      <td>subsystem396</td>\n",
       "      <td>UBE2A</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>subsystem396</td>\n",
       "      <td>UBE2B</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>UBE2I</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>400</th>\n",
       "      <td>subsystem140</td>\n",
       "      <td>UBE2N</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>UBE2T</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>subsystem140</td>\n",
       "      <td>UBE2V2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403</th>\n",
       "      <td>subsystem201</td>\n",
       "      <td>UBR7</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>UIMC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>subsystem29</td>\n",
       "      <td>UNG</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>USP1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>subsystem302</td>\n",
       "      <td>USP42</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>subsystem364</td>\n",
       "      <td>USP7</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>UVSSA</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>subsystem235</td>\n",
       "      <td>VRK1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>subsystem334</td>\n",
       "      <td>WDR4</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>subsystem209</td>\n",
       "      <td>WDR48</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>413</th>\n",
       "      <td>subsystem111</td>\n",
       "      <td>WRAP53</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>WRN</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>subsystem381</td>\n",
       "      <td>XAB2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>XAB2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>XPA</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>XPC</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>subsystem371</td>\n",
       "      <td>XRCC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>XRCC2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>subsystem58</td>\n",
       "      <td>XRCC2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>XRCC3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>subsystem297</td>\n",
       "      <td>XRCC4</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>XRCC4</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>subsystem370</td>\n",
       "      <td>XRCC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>XRCC6</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>427 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Parent         Child      EdgeType\n",
       "0    subsystem111  subsystem341  Child-Parent\n",
       "1    subsystem171  subsystem364  Child-Parent\n",
       "2    subsystem180  subsystem111  Child-Parent\n",
       "3    subsystem180  subsystem341  Child-Parent\n",
       "4    subsystem193  subsystem171  Child-Parent\n",
       "5    subsystem207  subsystem341  Child-Parent\n",
       "6    subsystem230  subsystem341  Child-Parent\n",
       "7    subsystem232  subsystem187  Child-Parent\n",
       "8    subsystem232  subsystem201  Child-Parent\n",
       "9    subsystem232  subsystem283  Child-Parent\n",
       "10   subsystem232  subsystem341  Child-Parent\n",
       "11   subsystem283  subsystem180  Child-Parent\n",
       "12   subsystem283  subsystem230  Child-Parent\n",
       "13   subsystem283  subsystem341  Child-Parent\n",
       "14   subsystem284  subsystem235  Child-Parent\n",
       "15   subsystem284  subsystem240  Child-Parent\n",
       "16   subsystem284  subsystem341  Child-Parent\n",
       "17   subsystem319  subsystem341  Child-Parent\n",
       "18   subsystem333  subsystem289  Child-Parent\n",
       "19   subsystem333  subsystem341  Child-Parent\n",
       "20   subsystem334  subsystem193  Child-Parent\n",
       "21   subsystem334  subsystem284  Child-Parent\n",
       "22   subsystem334  subsystem341  Child-Parent\n",
       "23   subsystem341   subsystem13  Child-Parent\n",
       "24   subsystem341   subsystem17  Child-Parent\n",
       "25   subsystem341   subsystem29  Child-Parent\n",
       "26   subsystem341   subsystem51  Child-Parent\n",
       "27   subsystem341   subsystem53  Child-Parent\n",
       "28   subsystem341   subsystem58  Child-Parent\n",
       "29   subsystem341   subsystem63  Child-Parent\n",
       "..            ...           ...           ...\n",
       "397  subsystem396         UBE2A     Gene-Term\n",
       "398  subsystem396         UBE2B     Gene-Term\n",
       "399  subsystem341         UBE2I     Gene-Term\n",
       "400  subsystem140         UBE2N     Gene-Term\n",
       "401  subsystem341         UBE2T     Gene-Term\n",
       "402  subsystem140        UBE2V2     Gene-Term\n",
       "403  subsystem201          UBR7     Gene-Term\n",
       "404  subsystem341         UIMC1     Gene-Term\n",
       "405   subsystem29           UNG     Gene-Term\n",
       "406  subsystem341          USP1     Gene-Term\n",
       "407  subsystem302         USP42     Gene-Term\n",
       "408  subsystem364          USP7     Gene-Term\n",
       "409  subsystem341         UVSSA     Gene-Term\n",
       "410  subsystem235          VRK1     Gene-Term\n",
       "411  subsystem334          WDR4     Gene-Term\n",
       "412  subsystem209         WDR48     Gene-Term\n",
       "413  subsystem111        WRAP53     Gene-Term\n",
       "414  subsystem341           WRN     Gene-Term\n",
       "415  subsystem381          XAB2     Gene-Term\n",
       "416  subsystem341          XAB2     Gene-Term\n",
       "417  subsystem341           XPA     Gene-Term\n",
       "418  subsystem341           XPC     Gene-Term\n",
       "419  subsystem371         XRCC1     Gene-Term\n",
       "420  subsystem341         XRCC2     Gene-Term\n",
       "421   subsystem58         XRCC2     Gene-Term\n",
       "422  subsystem341         XRCC3     Gene-Term\n",
       "423  subsystem297         XRCC4     Gene-Term\n",
       "424  subsystem341         XRCC4     Gene-Term\n",
       "425  subsystem370         XRCC5     Gene-Term\n",
       "426  subsystem341         XRCC6     Gene-Term\n",
       "\n",
       "[427 rows x 3 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open ('resultNew/co-expression.save', 'rb') as fp:\n",
    "    itemlist = pickle.load(fp)\n",
    "\n",
    "hierarchy  = [x for x in itemlist if \"subsystem\" in str(x[0])]\n",
    "mapping = [x for x in itemlist if \"subsystem\" not in str(x[0])]\n",
    "\n",
    "ont = Ontology(hierarchy, mapping)\n",
    "\n",
    "# Prints a summary of the ontology's structure\n",
    "print(ont)\n",
    "ont.to_table('toy_ontology3.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "437 genes, 74 terms, 533 gene-term relations, 74 term-term relations\n",
      "node_attributes: []\n",
      "edge_attributes: [2]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Alternative mechanism</td>\n",
       "      <td>MRN Complex</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Chromatin remodelling</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Chromosome segregation</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Modulation of nucleotide pools</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>p53 pathway</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Telomere maintenance</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>TLS</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Topisomerase damage reversal</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Ubiquitin response</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>BER</td>\n",
       "      <td>AP endonucleases</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>BER</td>\n",
       "      <td>DNA glycosylases</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>BER</td>\n",
       "      <td>LONG PATCH-BER factors</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>BER</td>\n",
       "      <td>Other BER factors</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>BER</td>\n",
       "      <td>SHORT PATCH-BER factors</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>damage in S phase</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>FPC (fork protection complex)</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>G1-CC phase</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>G1-S checkpoint</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>G2-CC phase</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>G2-M checkpoint</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>HRAD17(Rad24)- RFC complex</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>Rad17-Mec3- Ddc1 complex</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>RAD9-Hus1-Rad1 complex</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>S-CC phase</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>DSR</td>\n",
       "      <td>Alt-NHEJ</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>DSR</td>\n",
       "      <td>FA (Fanconi anemia pathway)</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>DSR</td>\n",
       "      <td>HR (Homologous Recombination)</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>DSR</td>\n",
       "      <td>NHEJ</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>HR (Homologous Recombination)</td>\n",
       "      <td>UIMC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>DNA glycosylases</td>\n",
       "      <td>UNG</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>UPF1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>FA (Fanconi anemia pathway)</td>\n",
       "      <td>USP1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>TLS</td>\n",
       "      <td>USP1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>UVRAG</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>TCR (Transcription coupled repair)</td>\n",
       "      <td>UVSSA</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>584</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>VCP</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>FA (Fanconi anemia pathway)</td>\n",
       "      <td>WDR48</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>TLS</td>\n",
       "      <td>WDR48</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>G1-CC phase</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>G2-CC phase</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>Telomere maintenance</td>\n",
       "      <td>WRAP53</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>LONG PATCH-BER factors</td>\n",
       "      <td>WRN</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>591</th>\n",
       "      <td>HR (Homologous Recombination)</td>\n",
       "      <td>WRN</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>592</th>\n",
       "      <td>TCR (Transcription coupled repair)</td>\n",
       "      <td>XAB2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>593</th>\n",
       "      <td>TCR (Transcription coupled repair)</td>\n",
       "      <td>XPA</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>594</th>\n",
       "      <td>GGR (Global genome repair)</td>\n",
       "      <td>XPC</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>595</th>\n",
       "      <td>SHORT PATCH-BER factors</td>\n",
       "      <td>XRCC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>TCR (Transcription coupled repair)</td>\n",
       "      <td>XRCC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>597</th>\n",
       "      <td>Alt-NHEJ</td>\n",
       "      <td>XRCC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>598</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC4</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>601</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC6</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>603</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>XRCC6BP1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>604</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>YBX1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>605</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>ZRANB3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>606</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>ZSWIM7</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>607 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Parent                           Child  \\\n",
       "0                 Alternative mechanism                     MRN Complex   \n",
       "1                    Associated process              Checkpoint factors   \n",
       "2                    Associated process           Chromatin remodelling   \n",
       "3                    Associated process          Chromosome segregation   \n",
       "4                    Associated process                 DNA replication   \n",
       "5                    Associated process  Modulation of nucleotide pools   \n",
       "6                    Associated process                     p53 pathway   \n",
       "7                    Associated process            Telomere maintenance   \n",
       "8                    Associated process                             TLS   \n",
       "9                    Associated process    Topisomerase damage reversal   \n",
       "10                   Associated process              Ubiquitin response   \n",
       "11                                  BER                AP endonucleases   \n",
       "12                                  BER                DNA glycosylases   \n",
       "13                                  BER          LONG PATCH-BER factors   \n",
       "14                                  BER               Other BER factors   \n",
       "15                                  BER         SHORT PATCH-BER factors   \n",
       "16                   Checkpoint factors               damage in S phase   \n",
       "17                   Checkpoint factors   FPC (fork protection complex)   \n",
       "18                   Checkpoint factors                     G1-CC phase   \n",
       "19                   Checkpoint factors                 G1-S checkpoint   \n",
       "20                   Checkpoint factors                     G2-CC phase   \n",
       "21                   Checkpoint factors                 G2-M checkpoint   \n",
       "22                   Checkpoint factors      HRAD17(Rad24)- RFC complex   \n",
       "23                   Checkpoint factors        Rad17-Mec3- Ddc1 complex   \n",
       "24                   Checkpoint factors          RAD9-Hus1-Rad1 complex   \n",
       "25                   Checkpoint factors                      S-CC phase   \n",
       "26                                  DSR                        Alt-NHEJ   \n",
       "27                                  DSR     FA (Fanconi anemia pathway)   \n",
       "28                                  DSR   HR (Homologous Recombination)   \n",
       "29                                  DSR                            NHEJ   \n",
       "..                                  ...                             ...   \n",
       "577       HR (Homologous Recombination)                           UIMC1   \n",
       "578                    DNA glycosylases                             UNG   \n",
       "579        Genes with probable DDR role                            UPF1   \n",
       "580         FA (Fanconi anemia pathway)                            USP1   \n",
       "581                                 TLS                            USP1   \n",
       "582        Genes with probable DDR role                           UVRAG   \n",
       "583  TCR (Transcription coupled repair)                           UVSSA   \n",
       "584        Genes with probable DDR role                             VCP   \n",
       "585         FA (Fanconi anemia pathway)                           WDR48   \n",
       "586                                 TLS                           WDR48   \n",
       "587                         G1-CC phase                            WEE1   \n",
       "588                         G2-CC phase                            WEE1   \n",
       "589                Telomere maintenance                          WRAP53   \n",
       "590              LONG PATCH-BER factors                             WRN   \n",
       "591       HR (Homologous Recombination)                             WRN   \n",
       "592  TCR (Transcription coupled repair)                            XAB2   \n",
       "593  TCR (Transcription coupled repair)                             XPA   \n",
       "594          GGR (Global genome repair)                             XPC   \n",
       "595             SHORT PATCH-BER factors                           XRCC1   \n",
       "596  TCR (Transcription coupled repair)                           XRCC1   \n",
       "597                            Alt-NHEJ                           XRCC1   \n",
       "598                                NHEJ                           XRCC2   \n",
       "599                                NHEJ                           XRCC3   \n",
       "600                                NHEJ                           XRCC4   \n",
       "601                                NHEJ                           XRCC5   \n",
       "602                                NHEJ                           XRCC6   \n",
       "603        Genes with probable DDR role                        XRCC6BP1   \n",
       "604        Genes with probable DDR role                            YBX1   \n",
       "605        Genes with probable DDR role                          ZRANB3   \n",
       "606        Genes with probable DDR role                          ZSWIM7   \n",
       "\n",
       "         EdgeType  \n",
       "0    Child-Parent  \n",
       "1    Child-Parent  \n",
       "2    Child-Parent  \n",
       "3    Child-Parent  \n",
       "4    Child-Parent  \n",
       "5    Child-Parent  \n",
       "6    Child-Parent  \n",
       "7    Child-Parent  \n",
       "8    Child-Parent  \n",
       "9    Child-Parent  \n",
       "10   Child-Parent  \n",
       "11   Child-Parent  \n",
       "12   Child-Parent  \n",
       "13   Child-Parent  \n",
       "14   Child-Parent  \n",
       "15   Child-Parent  \n",
       "16   Child-Parent  \n",
       "17   Child-Parent  \n",
       "18   Child-Parent  \n",
       "19   Child-Parent  \n",
       "20   Child-Parent  \n",
       "21   Child-Parent  \n",
       "22   Child-Parent  \n",
       "23   Child-Parent  \n",
       "24   Child-Parent  \n",
       "25   Child-Parent  \n",
       "26   Child-Parent  \n",
       "27   Child-Parent  \n",
       "28   Child-Parent  \n",
       "29   Child-Parent  \n",
       "..            ...  \n",
       "577     Gene-Term  \n",
       "578     Gene-Term  \n",
       "579     Gene-Term  \n",
       "580     Gene-Term  \n",
       "581     Gene-Term  \n",
       "582     Gene-Term  \n",
       "583     Gene-Term  \n",
       "584     Gene-Term  \n",
       "585     Gene-Term  \n",
       "586     Gene-Term  \n",
       "587     Gene-Term  \n",
       "588     Gene-Term  \n",
       "589     Gene-Term  \n",
       "590     Gene-Term  \n",
       "591     Gene-Term  \n",
       "592     Gene-Term  \n",
       "593     Gene-Term  \n",
       "594     Gene-Term  \n",
       "595     Gene-Term  \n",
       "596     Gene-Term  \n",
       "597     Gene-Term  \n",
       "598     Gene-Term  \n",
       "599     Gene-Term  \n",
       "600     Gene-Term  \n",
       "601     Gene-Term  \n",
       "602     Gene-Term  \n",
       "603     Gene-Term  \n",
       "604     Gene-Term  \n",
       "605     Gene-Term  \n",
       "606     Gene-Term  \n",
       "\n",
       "[607 rows x 3 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont0 = pd.read_csv('data/pearl2015.txt', sep='\\t', header=None)\n",
    "ont2 = Ontology.from_table(ont0)\n",
    "print(ont2)\n",
    "ont2.to_table('pearl2015.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'0\\n1\\n2\\n3\\n4\\n5\\n6\\n7\\n8\\n9\\n10\\n11\\n12\\n13\\n14\\n15\\n16\\n17\\n18\\n19\\n20\\n21\\n22\\n23\\n24\\n25\\n26\\n27\\n28\\n29\\n30\\n31\\n32\\n33\\n34\\n35\\n36\\n37\\n38\\n39\\n40\\n41\\n42\\n43\\n44\\n45\\n46\\n47\\n48\\n49\\n50\\n51\\n52\\n53\\n54\\n55\\n56\\n57\\n58\\n59\\n60\\n61\\n62\\n63\\n64\\n65\\n66\\n67\\n68\\n69\\n70\\n71\\n72\\n73\\n74\\n75\\n76\\n77\\n78\\n79\\n80\\n81\\n82\\n83\\n84\\n85\\n86\\n87\\n88\\n89\\n90\\n91\\n92\\n93\\n94\\n95\\n96\\n97\\n98\\n99\\n'\n"
     ]
    }
   ],
   "source": [
    "import subprocess\n",
    "args = (\"../../alignOntology/calculateFDRs\", \"toy_ontology2.txt\", \"pearl2015.txt\", \"0.01\", \"criss_cross\", \"testResult\", \"100\", \"30\", \"Gene-Term\")\n",
    "#Or just:\n",
    "#args = \"bin/bar -c somefile.xml -d text.txt -r aString -f anotherString\".split()\n",
    "popen = subprocess.Popen(args, stdout=subprocess.PIPE)\n",
    "popen.wait()\n",
    "output = popen.stdout.read()\n",
    "print(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "collapse command: /home/yan30/.ddot/lib/python3.7/site-packages/ddot/alignOntology/collapseRedundantNodes /tmp/tmphv1xevmv\n",
      "collapse command: /home/yan30/.ddot/lib/python3.7/site-packages/ddot/alignOntology/collapseRedundantNodes /tmp/tmprnvg1er_\n",
      "Alignment command: /home/yan30/.ddot/lib/python3.7/site-packages/ddot/alignOntology/calculateFDRs /tmp/tmp_i441__2 /tmp/tmpw8mq9kvt 0.05 criss_cross /tmp/tmp8gz_9o8z 100 64 gene\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Term</th>\n",
       "      <th>Similarity</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Term</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>subsystem37</th>\n",
       "      <td>Ubiquitins</td>\n",
       "      <td>0.444618</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem31</th>\n",
       "      <td>SUMO</td>\n",
       "      <td>0.444054</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem10</th>\n",
       "      <td>Alt-NHEJ</td>\n",
       "      <td>0.439860</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem396</th>\n",
       "      <td>SSR</td>\n",
       "      <td>0.297021</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem269</th>\n",
       "      <td>TLS</td>\n",
       "      <td>0.231794</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem391</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>0.189705</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem451</th>\n",
       "      <td>BER</td>\n",
       "      <td>0.183639</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem289</th>\n",
       "      <td>MMR</td>\n",
       "      <td>0.182943</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem456</th>\n",
       "      <td>DSR</td>\n",
       "      <td>0.160267</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem272</th>\n",
       "      <td>NER</td>\n",
       "      <td>0.129968</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem96</th>\n",
       "      <td>RFC (replication factor C)</td>\n",
       "      <td>0.256521</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem95</th>\n",
       "      <td>UBL-specific proteases (ULPs)</td>\n",
       "      <td>0.201562</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem8</th>\n",
       "      <td>FPC (fork protection complex)</td>\n",
       "      <td>0.881386</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem805</th>\n",
       "      <td>Repair pathway</td>\n",
       "      <td>0.619082</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem7</th>\n",
       "      <td>MutS homologs specialized for meiosis</td>\n",
       "      <td>0.887586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem6</th>\n",
       "      <td>MRN Complex</td>\n",
       "      <td>0.594000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem5</th>\n",
       "      <td>damage in S phase</td>\n",
       "      <td>0.881386</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem58</th>\n",
       "      <td>Riddle syndrome!</td>\n",
       "      <td>0.440850</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem52</th>\n",
       "      <td>Modulation of nucleotide pools</td>\n",
       "      <td>0.314578</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem4</th>\n",
       "      <td>Direct Repair (not in humans)</td>\n",
       "      <td>0.891414</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem49</th>\n",
       "      <td>epistasis group</td>\n",
       "      <td>0.296026</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem42</th>\n",
       "      <td>HRAD17(Rad24)- RFC complex</td>\n",
       "      <td>0.359540</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem40</th>\n",
       "      <td>RPA (replication factor A)</td>\n",
       "      <td>0.593273</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem401</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>0.499380</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem39</th>\n",
       "      <td>DNA polymerase epsilon</td>\n",
       "      <td>0.660657</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem36</th>\n",
       "      <td>UBL-conjugating enzymes</td>\n",
       "      <td>0.176977</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem35</th>\n",
       "      <td>RAD9-Hus1-Rad1 complex</td>\n",
       "      <td>0.359079</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem34</th>\n",
       "      <td>Rad17-Mec3- Ddc1 complex</td>\n",
       "      <td>0.446579</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem33</th>\n",
       "      <td>DNA replication</td>\n",
       "      <td>0.362578</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem2</th>\n",
       "      <td>S-CC phase</td>\n",
       "      <td>0.878101</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem29</th>\n",
       "      <td>MutL homologs</td>\n",
       "      <td>0.444021</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem290</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>0.173701</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem28</th>\n",
       "      <td>Ubiquitin- activating enzymes (E1)</td>\n",
       "      <td>0.425988</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem280</th>\n",
       "      <td>Ubiquitin response</td>\n",
       "      <td>0.186222</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem1</th>\n",
       "      <td>AP endonucleases</td>\n",
       "      <td>0.879606</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem194</th>\n",
       "      <td>Chromosome segregation</td>\n",
       "      <td>0.158264</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem18</th>\n",
       "      <td>Other MMR factors</td>\n",
       "      <td>0.447440</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem164</th>\n",
       "      <td>DNA glycosylases</td>\n",
       "      <td>0.171189</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem15</th>\n",
       "      <td>SHORT PATCH-BER factors</td>\n",
       "      <td>0.592860</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem157</th>\n",
       "      <td>Other single Ring- finger type E3</td>\n",
       "      <td>0.223494</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem156</th>\n",
       "      <td>Mismatch and loop recognition factors</td>\n",
       "      <td>0.787846</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem155</th>\n",
       "      <td>Ubiquitin- conjugating enzymes (E2)</td>\n",
       "      <td>0.423968</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem14</th>\n",
       "      <td>Ubiquitin-conjugating enzymes</td>\n",
       "      <td>0.908627</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem149</th>\n",
       "      <td>G1-S checkpoint</td>\n",
       "      <td>0.359365</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem134</th>\n",
       "      <td>p53 pathway</td>\n",
       "      <td>0.360663</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem12</th>\n",
       "      <td>G2-CC phase</td>\n",
       "      <td>0.592873</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem116</th>\n",
       "      <td>G2-M checkpoint</td>\n",
       "      <td>0.203809</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem0</th>\n",
       "      <td>Direct Repair</td>\n",
       "      <td>0.601929</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               Term  Similarity  FDR\n",
       "Term                                                                \n",
       "subsystem37                              Ubiquitins    0.444618  0.0\n",
       "subsystem31                                    SUMO    0.444054  0.0\n",
       "subsystem10                                Alt-NHEJ    0.439860  0.0\n",
       "subsystem396                                    SSR    0.297021  0.0\n",
       "subsystem269                                    TLS    0.231794  0.0\n",
       "subsystem391                                   NHEJ    0.189705  0.0\n",
       "subsystem451                                    BER    0.183639  0.0\n",
       "subsystem289                                    MMR    0.182943  0.0\n",
       "subsystem456                                    DSR    0.160267  0.0\n",
       "subsystem272                                    NER    0.129968  0.0\n",
       "subsystem96              RFC (replication factor C)    0.256521  0.0\n",
       "subsystem95           UBL-specific proteases (ULPs)    0.201562  0.0\n",
       "subsystem8            FPC (fork protection complex)    0.881386  0.0\n",
       "subsystem805                         Repair pathway    0.619082  0.0\n",
       "subsystem7    MutS homologs specialized for meiosis    0.887586  0.0\n",
       "subsystem6                              MRN Complex    0.594000  0.0\n",
       "subsystem5                        damage in S phase    0.881386  0.0\n",
       "subsystem58                        Riddle syndrome!    0.440850  0.0\n",
       "subsystem52          Modulation of nucleotide pools    0.314578  0.0\n",
       "subsystem4            Direct Repair (not in humans)    0.891414  0.0\n",
       "subsystem49                         epistasis group    0.296026  0.0\n",
       "subsystem42              HRAD17(Rad24)- RFC complex    0.359540  0.0\n",
       "subsystem40              RPA (replication factor A)    0.593273  0.0\n",
       "subsystem401                     Associated process    0.499380  0.0\n",
       "subsystem39                  DNA polymerase epsilon    0.660657  0.0\n",
       "subsystem36                 UBL-conjugating enzymes    0.176977  0.0\n",
       "subsystem35                  RAD9-Hus1-Rad1 complex    0.359079  0.0\n",
       "subsystem34                Rad17-Mec3- Ddc1 complex    0.446579  0.0\n",
       "subsystem33                         DNA replication    0.362578  0.0\n",
       "subsystem2                               S-CC phase    0.878101  0.0\n",
       "subsystem29                           MutL homologs    0.444021  0.0\n",
       "subsystem290                     Checkpoint factors    0.173701  0.0\n",
       "subsystem28      Ubiquitin- activating enzymes (E1)    0.425988  0.0\n",
       "subsystem280                     Ubiquitin response    0.186222  0.0\n",
       "subsystem1                         AP endonucleases    0.879606  0.0\n",
       "subsystem194                 Chromosome segregation    0.158264  0.0\n",
       "subsystem18                       Other MMR factors    0.447440  0.0\n",
       "subsystem164                       DNA glycosylases    0.171189  0.0\n",
       "subsystem15                 SHORT PATCH-BER factors    0.592860  0.0\n",
       "subsystem157      Other single Ring- finger type E3    0.223494  0.0\n",
       "subsystem156  Mismatch and loop recognition factors    0.787846  0.0\n",
       "subsystem155    Ubiquitin- conjugating enzymes (E2)    0.423968  0.0\n",
       "subsystem14           Ubiquitin-conjugating enzymes    0.908627  0.0\n",
       "subsystem149                        G1-S checkpoint    0.359365  0.0\n",
       "subsystem134                            p53 pathway    0.360663  0.0\n",
       "subsystem12                             G2-CC phase    0.592873  0.0\n",
       "subsystem116                        G2-M checkpoint    0.203809  0.0\n",
       "subsystem0                            Direct Repair    0.601929  0.0"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Reconstruct the ontology from the table\n",
    "ont.align(ont2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "https://test.ndexbio.org/v2/network/44c9f6d0-e27c-11ea-9101-0660b7976219\n"
     ]
    }
   ],
   "source": [
    "ndex_server = 'http://test.ndexbio.org' \n",
    "\n",
    "# Set the NDEx server and the user account (replace with your own account)\n",
    "ndex_user, ndex_pass = 'xiaoran.a.yan', 'a1l9i8f4e3'\n",
    "url, _ = ont.to_ndex(ndex_server=ndex_server, ndex_user=ndex_user, ndex_pass=ndex_pass)\n",
    "print(url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/home/yan30/.ddot/lib/python3.7/site-packages/logs/app.log'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mHTTPError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/site-packages/ndex/client.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, host, username, password, update_status, debug, user_agent)\u001b[0m\n\u001b[1;32m     57\u001b[0m                 \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mversion_url\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mheaders\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m'User-Agent'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0muserAgent\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0muser_agent\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 58\u001b[0;31m                 \u001b[0mresponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_for_status\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     59\u001b[0m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/site-packages/requests/models.py\u001b[0m in \u001b[0;36mraise_for_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    940\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mhttp_error_msg\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 941\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mHTTPError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhttp_error_msg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    942\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mHTTPError\u001b[0m: 404 Client Error: Not Found for url: https://test.ndexbio.org/rest/admin/status",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-22-999494a1fd2b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mont2\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mOntology\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfrom_ndex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mont2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/site-packages/ddot/Ontology.py\u001b[0m in \u001b[0;36mfrom_ndex\u001b[0;34m(cls, ndex_uuid, ndex_user, ndex_pass, ndex_server, edgetype_attr, edgetype_value)\u001b[0m\n\u001b[1;32m   1753\u001b[0m             \u001b[0musername\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mndex_user\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1754\u001b[0m             \u001b[0mpassword\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mndex_pass\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1755\u001b[0;31m             uuid=ndex_uuid)\n\u001b[0m\u001b[1;32m   1756\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1757\u001b[0m         return cls.from_NdexGraph(\n",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/site-packages/ndex/networkn.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, cx, server, username, password, uuid, networkx_G, data, user_agent, **attr)\u001b[0m\n\u001b[1;32m    187\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mcx\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mserver\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0muuid\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 189\u001b[0;31m             \u001b[0mndex\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mNdex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mserver\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0musername\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpassword\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0muser_agent\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0muser_agent\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    190\u001b[0m             \u001b[0mcx\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mndex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_network_as_cx_stream\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0muuid\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    191\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mcx\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/site-packages/ndex/client.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, host, username, password, update_status, debug, user_agent)\u001b[0m\n\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     75\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mreq_except\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mHTTPError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mhe\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 76\u001b[0;31m                 \u001b[0mndex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_logger\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'CLIENT'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwarning\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Can'\u001b[0m\u001b[0;34m't determine server version.'\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mhost\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m' Server returned error -- '\u001b[0m  \u001b[0;34m+\u001b[0m \u001b[0mhe\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmessage\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     77\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mversion\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"1.3\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     78\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhost\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhost\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\"/rest\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/site-packages/ndex/__init__.py\u001b[0m in \u001b[0;36mget_logger\u001b[0;34m(name, level)\u001b[0m\n\u001b[1;32m     18\u001b[0m     \u001b[0mformatter\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlogging\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mFormatter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'%(asctime)s.%(msecs)d '\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mname\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m' %(levelname)s: %(message)s'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'%Y-%m-%d %H:%M:%S'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     19\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 20\u001b[0;31m     \u001b[0mhandler\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlogging\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhandlers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTimedRotatingFileHandler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlog_path\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'app.log'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwhen\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'midnight'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbackupCount\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m28\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     21\u001b[0m     \u001b[0mhandler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetFormatter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mformatter\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     22\u001b[0m     \u001b[0mlogger\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maddHandler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandler\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/logging/handlers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, filename, when, interval, backupCount, encoding, delay, utc, atTime)\u001b[0m\n\u001b[1;32m    198\u001b[0m     \"\"\"\n\u001b[1;32m    199\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwhen\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'h'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minterval\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbackupCount\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencoding\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdelay\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mutc\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0matTime\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 200\u001b[0;31m         \u001b[0mBaseRotatingHandler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'a'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencoding\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdelay\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    201\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwhen\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwhen\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    202\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbackupCount\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbackupCount\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/logging/handlers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, filename, mode, encoding, delay)\u001b[0m\n\u001b[1;32m     53\u001b[0m         \u001b[0mUse\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mspecified\u001b[0m \u001b[0mfilename\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mstreamed\u001b[0m \u001b[0mlogging\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m         \"\"\"\n\u001b[0;32m---> 55\u001b[0;31m         \u001b[0mlogging\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mFileHandler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencoding\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdelay\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     56\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmode\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencoding\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mencoding\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/logging/__init__.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, filename, mode, encoding, delay)\u001b[0m\n\u001b[1;32m   1085\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstream\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1086\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1087\u001b[0;31m             \u001b[0mStreamHandler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1088\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1089\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/yan30/.ddot/lib/python3.7/logging/__init__.py\u001b[0m in \u001b[0;36m_open\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1114\u001b[0m         \u001b[0mReturn\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mresulting\u001b[0m \u001b[0mstream\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1115\u001b[0m         \"\"\"\n\u001b[0;32m-> 1116\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbaseFilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencoding\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencoding\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1117\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1118\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0memit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/home/yan30/.ddot/lib/python3.7/site-packages/logs/app.log'"
     ]
    }
   ],
   "source": [
    "ont2 = Ontology.from_ndex(url)\n",
    "print(ont2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "437"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G = ont2.to_networkx()\n",
    "#list(G.nodes(data=True))\n",
    "geneNodes = [x for x,y in G.nodes(data=True) if y['NodeType']=='Gene']\n",
    "GG = G.subgraph(geneNodes)\n",
    "len(GG.nodes())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'epistasis group': {'EdgeType': 'Gene-Term',\n",
       "  2: 'Gene-Term',\n",
       "  'Is_Tree_Edge': 'Not_Tree'},\n",
       " 'Genes with probable DDR role': {'EdgeType': 'Gene-Term',\n",
       "  2: 'Gene-Term',\n",
       "  'Is_Tree_Edge': 'Not_Tree'},\n",
       " 'Ubiquitin-conjugating enzymes': {'EdgeType': 'Gene-Term',\n",
       "  2: 'Gene-Term',\n",
       "  'Is_Tree_Edge': 'Tree'}}"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#len(G.nodes())\n",
    "G['UBE2V2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "import networkx as nx\n",
    "#import matplotlib.pyplot as plt\n",
    "\n",
    "graphs = [c for c in nx.connected_components(G.to_undirected())]\n",
    "print(len(graphs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['subsystem99',\n",
       " 'subsystem98',\n",
       " 'subsystem97',\n",
       " 'subsystem96',\n",
       " 'subsystem95',\n",
       " 'subsystem94',\n",
       " 'subsystem93',\n",
       " 'subsystem92',\n",
       " 'subsystem91',\n",
       " 'subsystem90',\n",
       " 'subsystem9',\n",
       " 'subsystem89',\n",
       " 'subsystem88',\n",
       " 'subsystem87',\n",
       " 'subsystem86',\n",
       " 'subsystem85',\n",
       " 'subsystem84',\n",
       " 'subsystem83',\n",
       " 'subsystem82',\n",
       " 'subsystem81',\n",
       " 'subsystem80',\n",
       " 'subsystem8',\n",
       " 'subsystem79',\n",
       " 'subsystem78',\n",
       " 'subsystem77',\n",
       " 'subsystem76',\n",
       " 'subsystem75',\n",
       " 'subsystem74',\n",
       " 'subsystem73',\n",
       " 'subsystem72',\n",
       " 'subsystem71',\n",
       " 'subsystem70',\n",
       " 'subsystem7',\n",
       " 'subsystem69',\n",
       " 'subsystem68',\n",
       " 'subsystem67',\n",
       " 'subsystem66',\n",
       " 'subsystem65',\n",
       " 'subsystem64',\n",
       " 'subsystem63',\n",
       " 'subsystem62',\n",
       " 'subsystem61',\n",
       " 'subsystem60',\n",
       " 'subsystem6',\n",
       " 'subsystem59',\n",
       " 'subsystem58',\n",
       " 'subsystem57',\n",
       " 'subsystem56',\n",
       " 'subsystem55',\n",
       " 'subsystem54',\n",
       " 'subsystem53',\n",
       " 'subsystem52',\n",
       " 'subsystem51',\n",
       " 'subsystem50',\n",
       " 'subsystem5',\n",
       " 'subsystem49',\n",
       " 'subsystem48',\n",
       " 'subsystem47',\n",
       " 'subsystem46',\n",
       " 'subsystem45',\n",
       " 'subsystem44',\n",
       " 'subsystem43',\n",
       " 'subsystem42',\n",
       " 'subsystem41',\n",
       " 'subsystem40',\n",
       " 'subsystem4',\n",
       " 'subsystem39',\n",
       " 'subsystem38',\n",
       " 'subsystem37',\n",
       " 'subsystem36',\n",
       " 'subsystem35',\n",
       " 'subsystem34',\n",
       " 'subsystem33',\n",
       " 'subsystem32',\n",
       " 'subsystem313',\n",
       " 'subsystem312',\n",
       " 'subsystem311',\n",
       " 'subsystem310',\n",
       " 'subsystem31',\n",
       " 'subsystem309',\n",
       " 'subsystem308',\n",
       " 'subsystem307',\n",
       " 'subsystem306',\n",
       " 'subsystem305',\n",
       " 'subsystem304',\n",
       " 'subsystem303',\n",
       " 'subsystem302',\n",
       " 'subsystem301',\n",
       " 'subsystem300',\n",
       " 'subsystem30',\n",
       " 'subsystem3',\n",
       " 'subsystem299',\n",
       " 'subsystem298',\n",
       " 'subsystem297',\n",
       " 'subsystem296',\n",
       " 'subsystem295',\n",
       " 'subsystem294',\n",
       " 'subsystem293',\n",
       " 'subsystem292',\n",
       " 'subsystem291',\n",
       " 'subsystem290',\n",
       " 'subsystem29',\n",
       " 'subsystem289',\n",
       " 'subsystem288',\n",
       " 'subsystem287',\n",
       " 'subsystem286',\n",
       " 'subsystem285',\n",
       " 'subsystem284',\n",
       " 'subsystem283',\n",
       " 'subsystem282',\n",
       " 'subsystem281',\n",
       " 'subsystem280',\n",
       " 'subsystem28',\n",
       " 'subsystem279',\n",
       " 'subsystem278',\n",
       " 'subsystem277',\n",
       " 'subsystem276',\n",
       " 'subsystem275',\n",
       " 'subsystem274',\n",
       " 'subsystem273',\n",
       " 'subsystem272',\n",
       " 'subsystem271',\n",
       " 'subsystem270',\n",
       " 'subsystem27',\n",
       " 'subsystem269',\n",
       " 'subsystem268',\n",
       " 'subsystem267',\n",
       " 'subsystem266',\n",
       " 'subsystem265',\n",
       " 'subsystem264',\n",
       " 'subsystem263',\n",
       " 'subsystem262',\n",
       " 'subsystem261',\n",
       " 'subsystem260',\n",
       " 'subsystem26',\n",
       " 'subsystem259',\n",
       " 'subsystem258',\n",
       " 'subsystem257',\n",
       " 'subsystem256',\n",
       " 'subsystem255',\n",
       " 'subsystem254',\n",
       " 'subsystem253',\n",
       " 'subsystem252',\n",
       " 'subsystem251',\n",
       " 'subsystem250',\n",
       " 'subsystem25',\n",
       " 'subsystem249',\n",
       " 'subsystem248',\n",
       " 'subsystem247',\n",
       " 'subsystem246',\n",
       " 'subsystem245',\n",
       " 'subsystem244',\n",
       " 'subsystem243',\n",
       " 'subsystem242',\n",
       " 'subsystem241',\n",
       " 'subsystem240',\n",
       " 'subsystem24',\n",
       " 'subsystem239',\n",
       " 'subsystem238',\n",
       " 'subsystem237',\n",
       " 'subsystem236',\n",
       " 'subsystem235',\n",
       " 'subsystem234',\n",
       " 'subsystem233',\n",
       " 'subsystem232',\n",
       " 'subsystem231',\n",
       " 'subsystem230',\n",
       " 'subsystem23',\n",
       " 'subsystem229',\n",
       " 'subsystem228',\n",
       " 'subsystem227',\n",
       " 'subsystem226',\n",
       " 'subsystem225',\n",
       " 'subsystem224',\n",
       " 'subsystem223',\n",
       " 'subsystem222',\n",
       " 'subsystem221',\n",
       " 'subsystem220',\n",
       " 'subsystem22',\n",
       " 'subsystem219',\n",
       " 'subsystem218',\n",
       " 'subsystem217',\n",
       " 'subsystem216',\n",
       " 'subsystem215',\n",
       " 'subsystem214',\n",
       " 'subsystem213',\n",
       " 'subsystem212',\n",
       " 'subsystem211',\n",
       " 'subsystem210',\n",
       " 'subsystem21',\n",
       " 'subsystem209',\n",
       " 'subsystem208',\n",
       " 'subsystem207',\n",
       " 'subsystem206',\n",
       " 'subsystem205',\n",
       " 'subsystem204',\n",
       " 'subsystem203',\n",
       " 'subsystem202',\n",
       " 'subsystem201',\n",
       " 'subsystem200',\n",
       " 'subsystem20',\n",
       " 'subsystem2',\n",
       " 'subsystem199',\n",
       " 'subsystem198',\n",
       " 'subsystem197',\n",
       " 'subsystem196',\n",
       " 'subsystem195',\n",
       " 'subsystem194',\n",
       " 'subsystem193',\n",
       " 'subsystem192',\n",
       " 'subsystem191',\n",
       " 'subsystem190',\n",
       " 'subsystem19',\n",
       " 'subsystem189',\n",
       " 'subsystem188',\n",
       " 'subsystem187',\n",
       " 'subsystem186',\n",
       " 'subsystem185',\n",
       " 'subsystem184',\n",
       " 'subsystem183',\n",
       " 'subsystem182',\n",
       " 'subsystem181',\n",
       " 'subsystem180',\n",
       " 'subsystem18',\n",
       " 'subsystem179',\n",
       " 'subsystem178',\n",
       " 'subsystem177',\n",
       " 'subsystem176',\n",
       " 'subsystem175',\n",
       " 'subsystem174',\n",
       " 'subsystem173',\n",
       " 'subsystem172',\n",
       " 'subsystem171',\n",
       " 'subsystem170',\n",
       " 'subsystem17',\n",
       " 'subsystem169',\n",
       " 'subsystem168',\n",
       " 'subsystem167',\n",
       " 'subsystem166',\n",
       " 'subsystem165',\n",
       " 'subsystem164',\n",
       " 'subsystem163',\n",
       " 'subsystem162',\n",
       " 'subsystem161',\n",
       " 'subsystem160',\n",
       " 'subsystem16',\n",
       " 'subsystem159',\n",
       " 'subsystem158',\n",
       " 'subsystem157',\n",
       " 'subsystem156',\n",
       " 'subsystem155',\n",
       " 'subsystem154',\n",
       " 'subsystem153',\n",
       " 'subsystem152',\n",
       " 'subsystem151',\n",
       " 'subsystem150',\n",
       " 'subsystem15',\n",
       " 'subsystem149',\n",
       " 'subsystem148',\n",
       " 'subsystem147',\n",
       " 'subsystem146',\n",
       " 'subsystem145',\n",
       " 'subsystem144',\n",
       " 'subsystem143',\n",
       " 'subsystem142',\n",
       " 'subsystem141',\n",
       " 'subsystem140',\n",
       " 'subsystem14',\n",
       " 'subsystem139',\n",
       " 'subsystem138',\n",
       " 'subsystem137',\n",
       " 'subsystem136',\n",
       " 'subsystem135',\n",
       " 'subsystem134',\n",
       " 'subsystem133',\n",
       " 'subsystem132',\n",
       " 'subsystem131',\n",
       " 'subsystem130',\n",
       " 'subsystem13',\n",
       " 'subsystem129',\n",
       " 'subsystem128',\n",
       " 'subsystem127',\n",
       " 'subsystem126',\n",
       " 'subsystem125',\n",
       " 'subsystem124',\n",
       " 'subsystem123',\n",
       " 'subsystem122',\n",
       " 'subsystem121',\n",
       " 'subsystem120',\n",
       " 'subsystem12',\n",
       " 'subsystem119',\n",
       " 'subsystem118',\n",
       " 'subsystem117',\n",
       " 'subsystem116',\n",
       " 'subsystem115',\n",
       " 'subsystem114',\n",
       " 'subsystem113',\n",
       " 'subsystem112',\n",
       " 'subsystem111',\n",
       " 'subsystem110',\n",
       " 'subsystem11',\n",
       " 'subsystem109',\n",
       " 'subsystem108',\n",
       " 'subsystem107',\n",
       " 'subsystem106',\n",
       " 'subsystem105',\n",
       " 'subsystem104',\n",
       " 'subsystem103',\n",
       " 'subsystem102',\n",
       " 'subsystem101',\n",
       " 'subsystem100',\n",
       " 'subsystem10',\n",
       " 'subsystem1',\n",
       " 'subsystem0',\n",
       " 'XRCC6',\n",
       " 'XRCC5',\n",
       " 'XRCC4',\n",
       " 'XRCC3',\n",
       " 'XRCC2',\n",
       " 'XRCC1',\n",
       " 'XPC',\n",
       " 'XPA',\n",
       " 'XAB2',\n",
       " 'WRN',\n",
       " 'WRAP53',\n",
       " 'WDR48',\n",
       " 'WDR4',\n",
       " 'VRK1',\n",
       " 'UVSSA',\n",
       " 'USP7',\n",
       " 'USP42',\n",
       " 'USP1',\n",
       " 'UNG',\n",
       " 'UIMC1',\n",
       " 'UBR7',\n",
       " 'UBE2V2',\n",
       " 'UBE2T',\n",
       " 'UBE2N',\n",
       " 'UBE2I',\n",
       " 'UBE2B',\n",
       " 'UBE2A',\n",
       " 'TUBGCP4',\n",
       " 'TTF1',\n",
       " 'TREX2',\n",
       " 'TREX1',\n",
       " 'TRA2B',\n",
       " 'TP53BP1',\n",
       " 'TP53',\n",
       " 'TOPBP1',\n",
       " 'TOP3A',\n",
       " 'TOP2B',\n",
       " 'TOP2A',\n",
       " 'TOP1',\n",
       " 'TONSL',\n",
       " 'TIPIN',\n",
       " 'TIMELESS',\n",
       " 'TICRR',\n",
       " 'TERF2',\n",
       " 'TELO2',\n",
       " 'TDP2',\n",
       " 'TDP1',\n",
       " 'TDG',\n",
       " 'SUPT16H',\n",
       " 'SUMO2',\n",
       " 'SSRP1',\n",
       " 'SPRTN',\n",
       " 'SPO11',\n",
       " 'SPATA5L1',\n",
       " 'SPATA5',\n",
       " 'SMUG1',\n",
       " 'SMC6',\n",
       " 'SMC4',\n",
       " 'SMC2',\n",
       " 'SMC1A',\n",
       " 'SMARCAD1',\n",
       " 'SMARCA5',\n",
       " 'SLX4',\n",
       " 'SLX1B',\n",
       " 'SLX1A',\n",
       " 'SLF1',\n",
       " 'SHPRH',\n",
       " 'SETMAR',\n",
       " 'SENP1',\n",
       " 'SEM1',\n",
       " 'RRP9',\n",
       " 'RRM2B',\n",
       " 'RPA4',\n",
       " 'RPA3',\n",
       " 'RPA2',\n",
       " 'RPA1',\n",
       " 'RNF8',\n",
       " 'RNF4',\n",
       " 'RNF168',\n",
       " 'RMI2',\n",
       " 'RMI1',\n",
       " 'RIF1',\n",
       " 'RFWD3',\n",
       " 'RFC5',\n",
       " 'RFC4',\n",
       " 'RFC3',\n",
       " 'RFC2',\n",
       " 'RFC1',\n",
       " 'REV3L',\n",
       " 'REV1',\n",
       " 'RECQL5',\n",
       " 'RECQL4',\n",
       " 'RECQL',\n",
       " 'RDM1',\n",
       " 'RBBP8',\n",
       " 'RAD9A',\n",
       " 'RAD54L',\n",
       " 'RAD54B',\n",
       " 'RAD52',\n",
       " 'RAD51D',\n",
       " 'RAD51C',\n",
       " 'RAD51B',\n",
       " 'RAD51AP1',\n",
       " 'RAD51',\n",
       " 'RAD50',\n",
       " 'RAD23B',\n",
       " 'RAD23A',\n",
       " 'RAD18',\n",
       " 'RAD17',\n",
       " 'RAD1',\n",
       " 'PRPF19',\n",
       " 'PRKDC',\n",
       " 'PRIM2',\n",
       " 'PRIM1',\n",
       " 'PPAT',\n",
       " 'POT1',\n",
       " 'POLR2G',\n",
       " 'POLR2D',\n",
       " 'POLQ',\n",
       " 'POLN',\n",
       " 'POLM',\n",
       " 'POLL',\n",
       " 'POLK',\n",
       " 'POLI',\n",
       " 'POLH',\n",
       " 'POLG',\n",
       " 'POLE3',\n",
       " 'POLE2',\n",
       " 'POLE',\n",
       " 'POLD4',\n",
       " 'POLD3',\n",
       " 'POLD2',\n",
       " 'POLD1',\n",
       " 'POLB',\n",
       " 'POLA2',\n",
       " 'POLA1',\n",
       " 'PNKP',\n",
       " 'PMS2',\n",
       " 'PMS1',\n",
       " 'PER1',\n",
       " 'PDS5B',\n",
       " 'PDS5A',\n",
       " 'PCNA',\n",
       " 'PAXIP1',\n",
       " 'PARP4',\n",
       " 'PARP3',\n",
       " 'PARP2',\n",
       " 'PARP1',\n",
       " 'PALB2',\n",
       " 'ORC6',\n",
       " 'ORC5',\n",
       " 'ORC2',\n",
       " 'ORC1',\n",
       " 'OGG1',\n",
       " 'NUP188',\n",
       " 'NUDT1',\n",
       " 'NTHL1',\n",
       " 'NHEJ1',\n",
       " 'NEIL3',\n",
       " 'NEIL2',\n",
       " 'NEIL1',\n",
       " 'NCAPH2',\n",
       " 'NCAPH',\n",
       " 'NCAPG2',\n",
       " 'NCAPG',\n",
       " 'NCAPD3',\n",
       " 'NCAPD2',\n",
       " 'NBN',\n",
       " 'NARFL',\n",
       " 'NABP2',\n",
       " 'MUTYH',\n",
       " 'MUS81',\n",
       " 'MTBP',\n",
       " 'MSH6',\n",
       " 'MSH5',\n",
       " 'MSH4',\n",
       " 'MSH3',\n",
       " 'MSH2',\n",
       " 'MRE11',\n",
       " 'MPLKIP',\n",
       " 'MPG',\n",
       " 'MNAT1',\n",
       " 'MMS22L',\n",
       " 'MMS19',\n",
       " 'MLH3',\n",
       " 'MLH1',\n",
       " 'MGMT',\n",
       " 'MDC1',\n",
       " 'MCMBP',\n",
       " 'MCM8',\n",
       " 'MCM7',\n",
       " 'MCM6',\n",
       " 'MCM5',\n",
       " 'MCM4',\n",
       " 'MCM3',\n",
       " 'MCM2',\n",
       " 'MCM10',\n",
       " 'MBD4',\n",
       " 'MAD2L2',\n",
       " 'LIG4',\n",
       " 'LIG3',\n",
       " 'LIG1',\n",
       " 'KIN',\n",
       " 'HUS1B',\n",
       " 'HUS1',\n",
       " 'HNRNPUL1',\n",
       " 'HNRNPR',\n",
       " 'HMGB2',\n",
       " 'HMGB1',\n",
       " 'HLTF',\n",
       " 'HELQ',\n",
       " 'H2AFX',\n",
       " 'GTF2H5',\n",
       " 'GTF2H4',\n",
       " 'GTF2H3',\n",
       " 'GTF2H2',\n",
       " 'GTF2H1',\n",
       " 'GMPS',\n",
       " 'GINS4',\n",
       " 'GINS3',\n",
       " 'GINS2',\n",
       " 'GINS1',\n",
       " 'GGCT',\n",
       " 'GEN1',\n",
       " 'GADD45G',\n",
       " 'GADD45A',\n",
       " 'FEN1',\n",
       " 'FANCM',\n",
       " 'FANCL',\n",
       " 'FANCI',\n",
       " 'FANCG',\n",
       " 'FANCF',\n",
       " 'FANCE',\n",
       " 'FANCD2',\n",
       " 'FANCC',\n",
       " 'FANCB',\n",
       " 'FANCA',\n",
       " 'FAN1',\n",
       " 'FAM175A',\n",
       " 'FAAP24',\n",
       " 'FAAP20',\n",
       " 'FAAP100',\n",
       " 'EXO1',\n",
       " 'ERCC8',\n",
       " 'ERCC6',\n",
       " 'ERCC5',\n",
       " 'ERCC4',\n",
       " 'ERCC3',\n",
       " 'ERCC2',\n",
       " 'ERCC1',\n",
       " 'ENDOV',\n",
       " 'EME2',\n",
       " 'EME1',\n",
       " 'DUT',\n",
       " 'DTL',\n",
       " 'DSCC1',\n",
       " 'DRG1',\n",
       " 'DNA2',\n",
       " 'DMC1',\n",
       " 'DHX9',\n",
       " 'DDB2',\n",
       " 'DDB1',\n",
       " 'DCLRE1C',\n",
       " 'DCLRE1B',\n",
       " 'DCLRE1A',\n",
       " 'DBF4B',\n",
       " 'CLSPN',\n",
       " 'CLK2',\n",
       " 'CKS2',\n",
       " 'CHTF18',\n",
       " 'CHEK2',\n",
       " 'CHEK1',\n",
       " 'CHAF1A',\n",
       " 'CETN3',\n",
       " 'CETN2',\n",
       " 'CEP152',\n",
       " 'CENPK',\n",
       " 'CDK7',\n",
       " 'CDK2',\n",
       " 'CDC7',\n",
       " 'CDC45',\n",
       " 'CCNH',\n",
       " 'CCNA2',\n",
       " 'CBX3',\n",
       " 'BRIP1',\n",
       " 'BRE',\n",
       " 'BRCC3',\n",
       " 'BRCA2',\n",
       " 'BRCA1',\n",
       " 'BRAT1',\n",
       " 'BLM',\n",
       " 'BCCIP',\n",
       " 'BARD1',\n",
       " 'ATRX',\n",
       " 'ATRIP',\n",
       " 'ATR',\n",
       " 'ATM',\n",
       " 'ATAD5',\n",
       " 'APTX',\n",
       " 'APLF',\n",
       " 'APEX2',\n",
       " 'APEX1',\n",
       " 'ALKBH3',\n",
       " 'ALKBH2',\n",
       " 'ADNP',\n",
       " 'ACD']"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted([n for n in G.out_degree()], reverse=True)  # degree sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALKBH2\n",
      "ATAD5\n",
      "ATR\n",
      "BRCC3\n",
      "CCNH\n",
      "CKS2\n",
      "CLK2\n",
      "DTL\n",
      "ERCC3\n",
      "ERCC5\n",
      "EXO1\n",
      "FANCD2\n",
      "FANCL\n",
      "GTF2H5\n",
      "MCM6\n",
      "MCM7\n",
      "NEIL1\n",
      "PCNA\n",
      "POT1\n",
      "RFC3\n",
      "RFWD3\n",
      "TP53\n",
      "subsystem101\n",
      "subsystem108\n",
      "subsystem110\n",
      "subsystem118\n",
      "subsystem134\n",
      "subsystem16\n",
      "subsystem176\n",
      "subsystem190\n",
      "subsystem196\n",
      "subsystem21\n",
      "subsystem210\n",
      "subsystem212\n",
      "subsystem228\n",
      "subsystem235\n",
      "subsystem247\n",
      "subsystem255\n",
      "subsystem263\n",
      "subsystem296\n",
      "subsystem62\n",
      "subsystem69\n",
      "subsystem76\n",
      "subsystem85\n",
      "subsystem97\n"
     ]
    }
   ],
   "source": [
    "outDegrees = G.out_degree()\n",
    "for node, oDegree in outDegrees.items():    # for name, age in dictionary.iteritems():  (for Python 2.x)\n",
    "    if oDegree > 1:\n",
    "        print(node)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "subsystem120\n",
      "subsystem232\n",
      "subsystem289\n",
      "subsystem290\n",
      "subsystem297\n",
      "subsystem311\n",
      "subsystem312\n"
     ]
    }
   ],
   "source": [
    "inDegrees = G.in_degree()\n",
    "for node, iDegree in inDegrees.items():    # for name, age in dictionary.iteritems():  (for Python 2.x)\n",
    "    if iDegree > 3:\n",
    "        print(node)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "Gorig = nx.read_edgelist(\"data/resnik.txt\", delimiter='\\t' , nodetype=str,  data=(('weight',float),))\n",
    "with open('hierarchyQ3clixo.save', 'wb') as fp:\n",
    "    pickle.dump(Gorig.edges(), fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{0: 'ACD',\n",
       " 1: 'ABL1',\n",
       " 2: 'ACTL6A',\n",
       " 3: 'ACTR5',\n",
       " 4: 'ACTR8',\n",
       " 5: 'ALKBH1',\n",
       " 6: 'ALKBH2',\n",
       " 7: 'ALKBH3',\n",
       " 8: 'AMN1',\n",
       " 9: 'ANKRD28',\n",
       " 10: 'ANKRD44',\n",
       " 11: 'ANKRD52',\n",
       " 12: 'APEX1',\n",
       " 13: 'APEX2',\n",
       " 14: 'APITD1',\n",
       " 15: 'APLF',\n",
       " 16: 'APTX',\n",
       " 17: 'ARID1A',\n",
       " 18: 'ARID1B',\n",
       " 19: 'ARID2',\n",
       " 20: 'ASF1A',\n",
       " 21: 'ASF1B',\n",
       " 22: 'ATF2',\n",
       " 23: 'ATM',\n",
       " 24: 'ATR',\n",
       " 25: 'ATRIP',\n",
       " 26: 'ATRX',\n",
       " 27: 'ATXN3',\n",
       " 28: 'BAP1',\n",
       " 29: 'BARD1',\n",
       " 30: 'BAZ1A',\n",
       " 31: 'BAZ1B',\n",
       " 32: 'BCAS2',\n",
       " 33: 'BLM',\n",
       " 34: 'BRCA1',\n",
       " 35: 'BRCA2',\n",
       " 36: 'BRCC3',\n",
       " 37: 'BRD7',\n",
       " 38: 'BRE',\n",
       " 39: 'BRIP1',\n",
       " 40: 'BTG2',\n",
       " 41: 'CCDC98',\n",
       " 42: 'CCNA1',\n",
       " 43: 'CCNA2',\n",
       " 44: 'CCNB1',\n",
       " 45: 'CCNB2',\n",
       " 46: 'CCNB3',\n",
       " 47: 'CCND1',\n",
       " 48: 'CCND2',\n",
       " 49: 'CCND3',\n",
       " 50: 'CCNE1',\n",
       " 51: 'CCNH',\n",
       " 52: 'CCNO',\n",
       " 53: 'CDC25A',\n",
       " 54: 'CDC25B',\n",
       " 55: 'CDC5L',\n",
       " 56: 'CDK2',\n",
       " 57: 'CDK4',\n",
       " 58: 'CDK7',\n",
       " 59: 'CDKN1A',\n",
       " 60: 'CDKN2A',\n",
       " 61: 'CDKN2D',\n",
       " 62: 'CETN2',\n",
       " 63: 'CHAF1A',\n",
       " 64: 'CHEK1',\n",
       " 65: 'CHEK2',\n",
       " 66: 'CHRAC1',\n",
       " 67: 'CIB1',\n",
       " 68: 'CLK2',\n",
       " 69: 'CLSPN',\n",
       " 70: 'CNOT7',\n",
       " 71: 'COPS2',\n",
       " 72: 'COPS3',\n",
       " 73: 'COPS4',\n",
       " 74: 'COPS5',\n",
       " 75: 'COPS6',\n",
       " 76: 'COPS7A',\n",
       " 77: 'COPS7B',\n",
       " 78: 'COPS8',\n",
       " 79: 'CRY1',\n",
       " 80: 'CRY2',\n",
       " 81: 'CSNK1D',\n",
       " 82: 'CSNK1E',\n",
       " 83: 'CTC1',\n",
       " 84: 'CUL3',\n",
       " 85: 'CUL4A',\n",
       " 86: 'CUL5',\n",
       " 87: 'DAXX',\n",
       " 88: 'DCLRE1A',\n",
       " 89: 'DCLRE1B',\n",
       " 90: 'DCLRE1C',\n",
       " 91: 'DDB1',\n",
       " 92: 'DDB2',\n",
       " 93: 'DDX11',\n",
       " 94: 'DKC1',\n",
       " 95: 'DLGAP5',\n",
       " 96: 'DMC1',\n",
       " 97: 'DNA2',\n",
       " 98: 'DNMT1',\n",
       " 99: 'DNTT',\n",
       " 100: 'DUT',\n",
       " 101: 'EID3',\n",
       " 102: 'EME1',\n",
       " 103: 'EME2',\n",
       " 104: 'ERCC1',\n",
       " 105: 'ERCC2',\n",
       " 106: 'ERCC3',\n",
       " 107: 'ERCC4',\n",
       " 108: 'ERCC5',\n",
       " 109: 'ERCC6',\n",
       " 110: 'ERCC8',\n",
       " 111: 'EXO1',\n",
       " 112: 'EYA3',\n",
       " 113: 'FAAP100',\n",
       " 114: 'FAAP24',\n",
       " 115: 'FAM175A',\n",
       " 116: 'FAN1',\n",
       " 117: 'FANCA',\n",
       " 118: 'FANCB',\n",
       " 119: 'FANCC',\n",
       " 120: 'FANCD2',\n",
       " 121: 'FANCE',\n",
       " 122: 'FANCG',\n",
       " 123: 'FANCI',\n",
       " 124: 'FANCL',\n",
       " 125: 'FANCM',\n",
       " 126: 'FBXO18',\n",
       " 127: 'FEN1',\n",
       " 128: 'GADD45A',\n",
       " 129: 'GADD45G',\n",
       " 130: 'GAR1',\n",
       " 131: 'GEN1',\n",
       " 132: 'GPS1',\n",
       " 133: 'GTF2H1',\n",
       " 134: 'GTF2H2',\n",
       " 135: 'GTF2H3',\n",
       " 136: 'GTF2H4',\n",
       " 137: 'GTF2H5',\n",
       " 138: 'H2AFX',\n",
       " 139: 'HELQ',\n",
       " 140: 'HES1',\n",
       " 141: 'HFM1',\n",
       " 142: 'HINFP',\n",
       " 143: 'HLTF',\n",
       " 144: 'HMGB1',\n",
       " 145: 'HMGB2',\n",
       " 146: 'HUS1',\n",
       " 147: 'HUS1B',\n",
       " 148: 'IGHMBP2',\n",
       " 149: 'INO80',\n",
       " 150: 'INO80C',\n",
       " 151: 'INO80D',\n",
       " 152: 'INO80E',\n",
       " 153: 'IP6K3',\n",
       " 154: 'KAT2A',\n",
       " 155: 'KAT5',\n",
       " 156: 'LIG1',\n",
       " 157: 'LIG3',\n",
       " 158: 'LIG4',\n",
       " 159: 'MAD2L2',\n",
       " 160: 'MBD4',\n",
       " 161: 'MCRS1',\n",
       " 162: 'MDC1',\n",
       " 163: 'MDM2',\n",
       " 164: 'MDM4',\n",
       " 165: 'MEN1',\n",
       " 166: 'MGMT',\n",
       " 167: 'MLH1',\n",
       " 168: 'MLH3',\n",
       " 169: 'MMS19',\n",
       " 170: 'MMS22L',\n",
       " 171: 'MNAT1',\n",
       " 172: 'MPG',\n",
       " 173: 'MPLKIP',\n",
       " 174: 'MRE11A',\n",
       " 175: 'MSH2',\n",
       " 176: 'MSH3',\n",
       " 177: 'MSH4',\n",
       " 178: 'MSH5',\n",
       " 179: 'MSH6',\n",
       " 180: 'MTOR',\n",
       " 181: 'MUS81',\n",
       " 182: 'MUTYH',\n",
       " 183: 'MVP',\n",
       " 184: 'MYBBP1A',\n",
       " 185: 'NABP1',\n",
       " 186: 'NABP2',\n",
       " 187: 'NBN',\n",
       " 188: 'NCAPD2',\n",
       " 189: 'NCAPD3',\n",
       " 190: 'NCAPG',\n",
       " 191: 'NCAPH',\n",
       " 192: 'NCAPH2',\n",
       " 193: 'NEIL1',\n",
       " 194: 'NEIL2',\n",
       " 195: 'NEIL3',\n",
       " 196: 'NFATC2IP',\n",
       " 197: 'NFRKB',\n",
       " 198: 'NHEJ1',\n",
       " 199: 'NHP2',\n",
       " 200: 'NOP10',\n",
       " 201: 'NPM1',\n",
       " 202: 'NSMCE1',\n",
       " 203: 'NSMCE2',\n",
       " 204: 'NSMCE4A',\n",
       " 205: 'NTHL1',\n",
       " 206: 'NUDT1',\n",
       " 207: 'OBFC1',\n",
       " 208: 'OBFC2B',\n",
       " 209: 'OGG1',\n",
       " 210: 'PALB2',\n",
       " 211: 'PAPD7',\n",
       " 212: 'PARG',\n",
       " 213: 'PARP1',\n",
       " 214: 'PARP3',\n",
       " 215: 'PARP4',\n",
       " 216: 'PARPBP',\n",
       " 217: 'PAXIP1',\n",
       " 218: 'PBRM1',\n",
       " 219: 'PCNA',\n",
       " 220: 'PDS5A',\n",
       " 221: 'PDS5B',\n",
       " 222: 'PER1',\n",
       " 223: 'PER2',\n",
       " 224: 'PER3',\n",
       " 225: 'PIAS1',\n",
       " 226: 'PIAS2',\n",
       " 227: 'PIAS3',\n",
       " 228: 'PIAS4',\n",
       " 229: 'PIF1',\n",
       " 230: 'PINX1',\n",
       " 231: 'PLRG1',\n",
       " 232: 'PMS1',\n",
       " 233: 'PMS2',\n",
       " 234: 'PMS2P5',\n",
       " 235: 'PNKP',\n",
       " 236: 'POLA1',\n",
       " 237: 'POLB',\n",
       " 238: 'POLD1',\n",
       " 239: 'POLD2',\n",
       " 240: 'POLD3',\n",
       " 241: 'POLD4',\n",
       " 242: 'POLE',\n",
       " 243: 'POLE2',\n",
       " 244: 'POLE3',\n",
       " 245: 'POLE4',\n",
       " 246: 'POLG',\n",
       " 247: 'POLG2',\n",
       " 248: 'POLH',\n",
       " 249: 'POLI',\n",
       " 250: 'POLK',\n",
       " 251: 'POLL',\n",
       " 252: 'POLM',\n",
       " 253: 'POLN',\n",
       " 254: 'POLQ',\n",
       " 255: 'POLR2A',\n",
       " 256: 'POLR2B',\n",
       " 257: 'POLR2C',\n",
       " 258: 'POLR2D',\n",
       " 259: 'POLR2E',\n",
       " 260: 'POLR2F',\n",
       " 261: 'POLR2G',\n",
       " 262: 'POLR2H',\n",
       " 263: 'POLR2I',\n",
       " 264: 'POLR2J',\n",
       " 265: 'POLR2K',\n",
       " 266: 'POLR2L',\n",
       " 267: 'POT1',\n",
       " 268: 'PPP4C',\n",
       " 269: 'PPP4R1',\n",
       " 270: 'PPP4R2',\n",
       " 271: 'PPP4R4',\n",
       " 272: 'PPP6C',\n",
       " 273: 'PPP6R1',\n",
       " 274: 'PPP6R2',\n",
       " 275: 'PPP6R3',\n",
       " 276: 'PRKCG',\n",
       " 277: 'PRKDC',\n",
       " 278: 'PRMT6',\n",
       " 279: 'RAD1',\n",
       " 280: 'RAD17',\n",
       " 281: 'RAD18',\n",
       " 282: 'RAD21',\n",
       " 283: 'RAD23A',\n",
       " 284: 'RAD23B',\n",
       " 285: 'RAD50',\n",
       " 286: 'RAD51',\n",
       " 287: 'RAD51B',\n",
       " 288: 'RAD51C',\n",
       " 289: 'RAD51D',\n",
       " 290: 'RAD52',\n",
       " 291: 'RAD54B',\n",
       " 292: 'RAD54L',\n",
       " 293: 'RAD9A',\n",
       " 294: 'RAD9B',\n",
       " 295: 'RASSF7',\n",
       " 296: 'RBBP8',\n",
       " 297: 'RBX1',\n",
       " 298: 'RDM1',\n",
       " 299: 'RECQL',\n",
       " 300: 'RECQL4',\n",
       " 301: 'RECQL5',\n",
       " 302: 'REV1',\n",
       " 303: 'REV3L',\n",
       " 304: 'RFC1',\n",
       " 305: 'RFC2',\n",
       " 306: 'RFC3',\n",
       " 307: 'RFC4',\n",
       " 308: 'RFC5',\n",
       " 309: 'RIF1',\n",
       " 310: 'RMI2',\n",
       " 311: 'RNF168',\n",
       " 312: 'RNF4',\n",
       " 313: 'RNF8',\n",
       " 314: 'RPA1',\n",
       " 315: 'RPA2',\n",
       " 316: 'RPA3',\n",
       " 317: 'RPA4',\n",
       " 318: 'RPAIN',\n",
       " 319: 'RPS27L',\n",
       " 320: 'RRM1',\n",
       " 321: 'RRM2',\n",
       " 322: 'RRM2B',\n",
       " 323: 'RTEL1',\n",
       " 324: 'RUVBL1',\n",
       " 325: 'RUVBL2',\n",
       " 326: 'SETMAR',\n",
       " 327: 'SETX',\n",
       " 328: 'SFR1',\n",
       " 329: 'SHFM1',\n",
       " 330: 'SHPRH',\n",
       " 331: 'SIRT1',\n",
       " 332: 'SLX1A',\n",
       " 333: 'SLX1B',\n",
       " 334: 'SLX4',\n",
       " 335: 'SMARCA2',\n",
       " 336: 'SMARCA4',\n",
       " 337: 'SMARCA5',\n",
       " 338: 'SMARCAD1',\n",
       " 339: 'SMARCAL1',\n",
       " 340: 'SMARCB1',\n",
       " 341: 'SMARCC1',\n",
       " 342: 'SMARCC2',\n",
       " 343: 'SMARCD1',\n",
       " 344: 'SMARCE1',\n",
       " 345: 'SMC1A',\n",
       " 346: 'SMC1B',\n",
       " 347: 'SMC2',\n",
       " 348: 'SMC3',\n",
       " 349: 'SMC4',\n",
       " 350: 'SMC5',\n",
       " 351: 'SMC6',\n",
       " 352: 'SMEK2',\n",
       " 353: 'SMG6',\n",
       " 354: 'SMUG1',\n",
       " 355: 'SOD1',\n",
       " 356: 'SPO11',\n",
       " 357: 'STAG1',\n",
       " 358: 'STAG2',\n",
       " 359: 'STRA13',\n",
       " 360: 'SUMO1',\n",
       " 361: 'SUMO2',\n",
       " 362: 'SUMO3',\n",
       " 363: 'SUMO4',\n",
       " 364: 'SWI5',\n",
       " 365: 'TADA3',\n",
       " 366: 'TCEA1',\n",
       " 367: 'TCEB1',\n",
       " 368: 'TCEB2',\n",
       " 369: 'TCEB3',\n",
       " 370: 'TDG',\n",
       " 371: 'TDP1',\n",
       " 372: 'TELO2',\n",
       " 373: 'TEN1',\n",
       " 374: 'TEP1',\n",
       " 375: 'TERF1',\n",
       " 376: 'TERF2',\n",
       " 377: 'TERF2IP',\n",
       " 378: 'TERT',\n",
       " 379: 'TFPT',\n",
       " 380: 'TIMELESS',\n",
       " 381: 'TINF2',\n",
       " 382: 'TIPIN',\n",
       " 383: 'TNKS',\n",
       " 384: 'TNP1',\n",
       " 385: 'TONSL',\n",
       " 386: 'TOP1',\n",
       " 387: 'TOP2A',\n",
       " 388: 'TOP2B',\n",
       " 389: 'TOP3A',\n",
       " 390: 'TOP3B',\n",
       " 391: 'TOPBP1',\n",
       " 392: 'TP53',\n",
       " 393: 'TP53BP1',\n",
       " 394: 'TP73',\n",
       " 395: 'TREX1',\n",
       " 396: 'TREX2',\n",
       " 397: 'TRIM28',\n",
       " 398: 'TTI1',\n",
       " 399: 'TTI2',\n",
       " 400: 'TYMS',\n",
       " 401: 'UBA1',\n",
       " 402: 'UBA2',\n",
       " 403: 'UBB',\n",
       " 404: 'UBC',\n",
       " 405: 'UBD',\n",
       " 406: 'UBE2A',\n",
       " 407: 'UBE2B',\n",
       " 408: 'UBE2I',\n",
       " 409: 'UBE2N',\n",
       " 410: 'UBE2T',\n",
       " 411: 'UBE2V1',\n",
       " 412: 'UBE2V2',\n",
       " 413: 'UIMC1',\n",
       " 414: 'UNG',\n",
       " 415: 'UPF1',\n",
       " 416: 'USP1',\n",
       " 417: 'UVRAG',\n",
       " 418: 'UVSSA',\n",
       " 419: 'VCP',\n",
       " 420: 'WDR48',\n",
       " 421: 'WEE1',\n",
       " 422: 'WRAP53',\n",
       " 423: 'WRN',\n",
       " 424: 'XAB2',\n",
       " 425: 'XPA',\n",
       " 426: 'XPC',\n",
       " 427: 'XRCC1',\n",
       " 428: 'XRCC2',\n",
       " 429: 'XRCC3',\n",
       " 430: 'XRCC4',\n",
       " 431: 'XRCC5',\n",
       " 432: 'XRCC6',\n",
       " 433: 'XRCC6BP1',\n",
       " 434: 'YBX1',\n",
       " 435: 'ZRANB3',\n",
       " 436: 'ZSWIM7'}"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from networkx import relabel\n",
    "\n",
    "sim = nx.adjacency_matrix(Gorig).todense()\n",
    "Gnew = relabel.convert_node_labels_to_integers(Gorig, label_attribute = \"term\") #initilaization\n",
    "terms = nx.get_node_attributes(Gnew, 'term')\n",
    "terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "437 genes, 59 terms, 516 gene-term relations, 66 term-term relations\n",
      "node_attributes: ['Parent weight']\n",
      "edge_attributes: ['CLIXO_score']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S:452</td>\n",
       "      <td>S:446</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S:462</td>\n",
       "      <td>S:446</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S:471</td>\n",
       "      <td>S:459</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S:478</td>\n",
       "      <td>S:462</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S:478</td>\n",
       "      <td>S:463</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:441</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:449</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:455</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:456</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:457</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:471</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>S:481</td>\n",
       "      <td>S:448</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:437</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:447</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:452</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:454</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:464</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:467</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>S:485</td>\n",
       "      <td>S:455</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:439</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:445</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:458</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:470</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:475</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:477</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>S:487</td>\n",
       "      <td>S:455</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>S:488</td>\n",
       "      <td>S:438</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>S:488</td>\n",
       "      <td>S:442</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>S:488</td>\n",
       "      <td>S:443</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>S:488</td>\n",
       "      <td>S:444</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>S:467</td>\n",
       "      <td>72</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553</th>\n",
       "      <td>S:467</td>\n",
       "      <td>73</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>554</th>\n",
       "      <td>S:467</td>\n",
       "      <td>74</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>555</th>\n",
       "      <td>S:467</td>\n",
       "      <td>75</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>556</th>\n",
       "      <td>S:467</td>\n",
       "      <td>76</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>S:467</td>\n",
       "      <td>77</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558</th>\n",
       "      <td>S:467</td>\n",
       "      <td>78</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>S:440</td>\n",
       "      <td>79</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>S:488</td>\n",
       "      <td>8</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>561</th>\n",
       "      <td>S:440</td>\n",
       "      <td>80</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>562</th>\n",
       "      <td>S:490</td>\n",
       "      <td>81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>563</th>\n",
       "      <td>S:490</td>\n",
       "      <td>82</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>S:481</td>\n",
       "      <td>83</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>S:485</td>\n",
       "      <td>84</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>566</th>\n",
       "      <td>S:466</td>\n",
       "      <td>85</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567</th>\n",
       "      <td>S:485</td>\n",
       "      <td>86</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>S:481</td>\n",
       "      <td>87</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>569</th>\n",
       "      <td>S:490</td>\n",
       "      <td>88</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>570</th>\n",
       "      <td>S:481</td>\n",
       "      <td>89</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>571</th>\n",
       "      <td>S:474</td>\n",
       "      <td>9</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>572</th>\n",
       "      <td>S:479</td>\n",
       "      <td>90</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>573</th>\n",
       "      <td>S:466</td>\n",
       "      <td>91</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>574</th>\n",
       "      <td>S:466</td>\n",
       "      <td>92</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>575</th>\n",
       "      <td>S:490</td>\n",
       "      <td>93</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>S:481</td>\n",
       "      <td>94</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>S:490</td>\n",
       "      <td>95</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>S:490</td>\n",
       "      <td>96</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>S:484</td>\n",
       "      <td>97</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>S:490</td>\n",
       "      <td>98</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>S:479</td>\n",
       "      <td>99</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>582 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Parent  Child      EdgeType\n",
       "0    S:452  S:446  Child-Parent\n",
       "1    S:462  S:446  Child-Parent\n",
       "2    S:471  S:459  Child-Parent\n",
       "3    S:478  S:462  Child-Parent\n",
       "4    S:478  S:463  Child-Parent\n",
       "5    S:480  S:441  Child-Parent\n",
       "6    S:480  S:449  Child-Parent\n",
       "7    S:480  S:455  Child-Parent\n",
       "8    S:480  S:456  Child-Parent\n",
       "9    S:480  S:457  Child-Parent\n",
       "10   S:480  S:471  Child-Parent\n",
       "11   S:481  S:448  Child-Parent\n",
       "12   S:483  S:437  Child-Parent\n",
       "13   S:483  S:447  Child-Parent\n",
       "14   S:483  S:452  Child-Parent\n",
       "15   S:483  S:454  Child-Parent\n",
       "16   S:483  S:464  Child-Parent\n",
       "17   S:483  S:467  Child-Parent\n",
       "18   S:485  S:455  Child-Parent\n",
       "19   S:486  S:439  Child-Parent\n",
       "20   S:486  S:445  Child-Parent\n",
       "21   S:486  S:458  Child-Parent\n",
       "22   S:486  S:470  Child-Parent\n",
       "23   S:486  S:475  Child-Parent\n",
       "24   S:486  S:477  Child-Parent\n",
       "25   S:487  S:455  Child-Parent\n",
       "26   S:488  S:438  Child-Parent\n",
       "27   S:488  S:442  Child-Parent\n",
       "28   S:488  S:443  Child-Parent\n",
       "29   S:488  S:444  Child-Parent\n",
       "..     ...    ...           ...\n",
       "552  S:467     72     Gene-Term\n",
       "553  S:467     73     Gene-Term\n",
       "554  S:467     74     Gene-Term\n",
       "555  S:467     75     Gene-Term\n",
       "556  S:467     76     Gene-Term\n",
       "557  S:467     77     Gene-Term\n",
       "558  S:467     78     Gene-Term\n",
       "559  S:440     79     Gene-Term\n",
       "560  S:488      8     Gene-Term\n",
       "561  S:440     80     Gene-Term\n",
       "562  S:490     81     Gene-Term\n",
       "563  S:490     82     Gene-Term\n",
       "564  S:481     83     Gene-Term\n",
       "565  S:485     84     Gene-Term\n",
       "566  S:466     85     Gene-Term\n",
       "567  S:485     86     Gene-Term\n",
       "568  S:481     87     Gene-Term\n",
       "569  S:490     88     Gene-Term\n",
       "570  S:481     89     Gene-Term\n",
       "571  S:474      9     Gene-Term\n",
       "572  S:479     90     Gene-Term\n",
       "573  S:466     91     Gene-Term\n",
       "574  S:466     92     Gene-Term\n",
       "575  S:490     93     Gene-Term\n",
       "576  S:481     94     Gene-Term\n",
       "577  S:490     95     Gene-Term\n",
       "578  S:490     96     Gene-Term\n",
       "579  S:484     97     Gene-Term\n",
       "580  S:490     98     Gene-Term\n",
       "581  S:479     99     Gene-Term\n",
       "\n",
       "[582 rows x 3 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont1 = Ontology.run_clixo(sim, alpha=0.0, beta=1.0, square=True, square_names=terms)\n",
    "print(ont1)\n",
    "ont1.to_table('toy_ontology3clixo.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>EdgeType</th>\n",
       "      <th>CLIXO_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S:452</td>\n",
       "      <td>S:446</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>6.77149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S:462</td>\n",
       "      <td>S:446</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>6.18653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S:471</td>\n",
       "      <td>S:459</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>5.44956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S:478</td>\n",
       "      <td>S:462</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>4.52356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S:478</td>\n",
       "      <td>S:463</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>4.52356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:441</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>4.07105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:449</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>4.07105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:455</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>4.07105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:456</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>4.07105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:457</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>4.07105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>S:480</td>\n",
       "      <td>S:471</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>4.07105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>S:481</td>\n",
       "      <td>S:448</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:437</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:447</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:452</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:454</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:464</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>S:483</td>\n",
       "      <td>S:467</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>S:485</td>\n",
       "      <td>S:455</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.41394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:439</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.41394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:445</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.41394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:458</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.41394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:470</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.41394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:475</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.41394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>S:486</td>\n",
       "      <td>S:477</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.41394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>S:487</td>\n",
       "      <td>S:455</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.12763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>S:488</td>\n",
       "      <td>S:438</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.04357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>S:488</td>\n",
       "      <td>S:442</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.04357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>S:488</td>\n",
       "      <td>S:443</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.04357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>S:488</td>\n",
       "      <td>S:444</td>\n",
       "      <td>Child-Parent</td>\n",
       "      <td>3.04357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>S:467</td>\n",
       "      <td>72</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.60156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553</th>\n",
       "      <td>S:467</td>\n",
       "      <td>73</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.60156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>554</th>\n",
       "      <td>S:467</td>\n",
       "      <td>74</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.60156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>555</th>\n",
       "      <td>S:467</td>\n",
       "      <td>75</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.60156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>556</th>\n",
       "      <td>S:467</td>\n",
       "      <td>76</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.60156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>S:467</td>\n",
       "      <td>77</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.60156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558</th>\n",
       "      <td>S:467</td>\n",
       "      <td>78</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.60156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>S:440</td>\n",
       "      <td>79</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>7.77149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>S:488</td>\n",
       "      <td>8</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>3.04357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>561</th>\n",
       "      <td>S:440</td>\n",
       "      <td>80</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>7.77149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>562</th>\n",
       "      <td>S:490</td>\n",
       "      <td>81</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>2.48609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>563</th>\n",
       "      <td>S:490</td>\n",
       "      <td>82</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>2.48609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>S:481</td>\n",
       "      <td>83</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>S:485</td>\n",
       "      <td>84</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>3.41394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>566</th>\n",
       "      <td>S:466</td>\n",
       "      <td>85</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567</th>\n",
       "      <td>S:485</td>\n",
       "      <td>86</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>3.41394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>S:481</td>\n",
       "      <td>87</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>569</th>\n",
       "      <td>S:490</td>\n",
       "      <td>88</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>2.48609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>570</th>\n",
       "      <td>S:481</td>\n",
       "      <td>89</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>571</th>\n",
       "      <td>S:474</td>\n",
       "      <td>9</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.31206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>572</th>\n",
       "      <td>S:479</td>\n",
       "      <td>90</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>4.12763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>573</th>\n",
       "      <td>S:466</td>\n",
       "      <td>91</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>574</th>\n",
       "      <td>S:466</td>\n",
       "      <td>92</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>5.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>575</th>\n",
       "      <td>S:490</td>\n",
       "      <td>93</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>2.48609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>S:481</td>\n",
       "      <td>94</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>3.96413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>S:490</td>\n",
       "      <td>95</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>2.48609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>S:490</td>\n",
       "      <td>96</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>2.48609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>S:484</td>\n",
       "      <td>97</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>3.64221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>S:490</td>\n",
       "      <td>98</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>2.48609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>S:479</td>\n",
       "      <td>99</td>\n",
       "      <td>Gene-Term</td>\n",
       "      <td>4.12763</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>582 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Parent  Child      EdgeType  CLIXO_score\n",
       "0    S:452  S:446  Child-Parent      6.77149\n",
       "1    S:462  S:446  Child-Parent      6.18653\n",
       "2    S:471  S:459  Child-Parent      5.44956\n",
       "3    S:478  S:462  Child-Parent      4.52356\n",
       "4    S:478  S:463  Child-Parent      4.52356\n",
       "5    S:480  S:441  Child-Parent      4.07105\n",
       "6    S:480  S:449  Child-Parent      4.07105\n",
       "7    S:480  S:455  Child-Parent      4.07105\n",
       "8    S:480  S:456  Child-Parent      4.07105\n",
       "9    S:480  S:457  Child-Parent      4.07105\n",
       "10   S:480  S:471  Child-Parent      4.07105\n",
       "11   S:481  S:448  Child-Parent      3.96413\n",
       "12   S:483  S:437  Child-Parent      3.96413\n",
       "13   S:483  S:447  Child-Parent      3.96413\n",
       "14   S:483  S:452  Child-Parent      3.96413\n",
       "15   S:483  S:454  Child-Parent      3.96413\n",
       "16   S:483  S:464  Child-Parent      3.96413\n",
       "17   S:483  S:467  Child-Parent      3.96413\n",
       "18   S:485  S:455  Child-Parent      3.41394\n",
       "19   S:486  S:439  Child-Parent      3.41394\n",
       "20   S:486  S:445  Child-Parent      3.41394\n",
       "21   S:486  S:458  Child-Parent      3.41394\n",
       "22   S:486  S:470  Child-Parent      3.41394\n",
       "23   S:486  S:475  Child-Parent      3.41394\n",
       "24   S:486  S:477  Child-Parent      3.41394\n",
       "25   S:487  S:455  Child-Parent      3.12763\n",
       "26   S:488  S:438  Child-Parent      3.04357\n",
       "27   S:488  S:442  Child-Parent      3.04357\n",
       "28   S:488  S:443  Child-Parent      3.04357\n",
       "29   S:488  S:444  Child-Parent      3.04357\n",
       "..     ...    ...           ...          ...\n",
       "552  S:467     72     Gene-Term      5.60156\n",
       "553  S:467     73     Gene-Term      5.60156\n",
       "554  S:467     74     Gene-Term      5.60156\n",
       "555  S:467     75     Gene-Term      5.60156\n",
       "556  S:467     76     Gene-Term      5.60156\n",
       "557  S:467     77     Gene-Term      5.60156\n",
       "558  S:467     78     Gene-Term      5.60156\n",
       "559  S:440     79     Gene-Term      7.77149\n",
       "560  S:488      8     Gene-Term      3.04357\n",
       "561  S:440     80     Gene-Term      7.77149\n",
       "562  S:490     81     Gene-Term      2.48609\n",
       "563  S:490     82     Gene-Term      2.48609\n",
       "564  S:481     83     Gene-Term      3.96413\n",
       "565  S:485     84     Gene-Term      3.41394\n",
       "566  S:466     85     Gene-Term      5.96413\n",
       "567  S:485     86     Gene-Term      3.41394\n",
       "568  S:481     87     Gene-Term      3.96413\n",
       "569  S:490     88     Gene-Term      2.48609\n",
       "570  S:481     89     Gene-Term      3.96413\n",
       "571  S:474      9     Gene-Term      5.31206\n",
       "572  S:479     90     Gene-Term      4.12763\n",
       "573  S:466     91     Gene-Term      5.96413\n",
       "574  S:466     92     Gene-Term      5.96413\n",
       "575  S:490     93     Gene-Term      2.48609\n",
       "576  S:481     94     Gene-Term      3.96413\n",
       "577  S:490     95     Gene-Term      2.48609\n",
       "578  S:490     96     Gene-Term      2.48609\n",
       "579  S:484     97     Gene-Term      3.64221\n",
       "580  S:490     98     Gene-Term      2.48609\n",
       "581  S:479     99     Gene-Term      4.12763\n",
       "\n",
       "[582 rows x 4 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont1.align(ont2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "collapse command: /home/yan30/.ddot/lib/python3.7/site-packages/ddot/alignOntology/collapseRedundantNodes /tmp/tmpgmne2q8l\n",
      "collapse command: /home/yan30/.ddot/lib/python3.7/site-packages/ddot/alignOntology/collapseRedundantNodes /tmp/tmpweblaj75\n",
      "Alignment command: /home/yan30/.ddot/lib/python3.7/site-packages/ddot/alignOntology/calculateFDRs /tmp/tmpc7bo7skg /tmp/tmpt30q8fvy 0.05 criss_cross /tmp/tmpq7fze9ur 100 64 gene\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Term</th>\n",
       "      <th>Similarity</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Term</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>subsystem37</th>\n",
       "      <td>Ubiquitins</td>\n",
       "      <td>0.444618</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem31</th>\n",
       "      <td>SUMO</td>\n",
       "      <td>0.444054</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem10</th>\n",
       "      <td>Alt-NHEJ</td>\n",
       "      <td>0.439860</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem396</th>\n",
       "      <td>SSR</td>\n",
       "      <td>0.297021</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem269</th>\n",
       "      <td>TLS</td>\n",
       "      <td>0.231794</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem391</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>0.189705</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem451</th>\n",
       "      <td>BER</td>\n",
       "      <td>0.183639</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem289</th>\n",
       "      <td>MMR</td>\n",
       "      <td>0.182943</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem456</th>\n",
       "      <td>DSR</td>\n",
       "      <td>0.160267</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem272</th>\n",
       "      <td>NER</td>\n",
       "      <td>0.129968</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem96</th>\n",
       "      <td>RFC (replication factor C)</td>\n",
       "      <td>0.256521</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem95</th>\n",
       "      <td>UBL-specific proteases (ULPs)</td>\n",
       "      <td>0.201562</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem8</th>\n",
       "      <td>FPC (fork protection complex)</td>\n",
       "      <td>0.881386</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem805</th>\n",
       "      <td>Repair pathway</td>\n",
       "      <td>0.619082</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem7</th>\n",
       "      <td>MutS homologs specialized for meiosis</td>\n",
       "      <td>0.887586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem6</th>\n",
       "      <td>MRN Complex</td>\n",
       "      <td>0.594000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem5</th>\n",
       "      <td>damage in S phase</td>\n",
       "      <td>0.881386</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem58</th>\n",
       "      <td>Riddle syndrome!</td>\n",
       "      <td>0.440850</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem52</th>\n",
       "      <td>Modulation of nucleotide pools</td>\n",
       "      <td>0.314578</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem4</th>\n",
       "      <td>Direct Repair (not in humans)</td>\n",
       "      <td>0.891414</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem49</th>\n",
       "      <td>epistasis group</td>\n",
       "      <td>0.296026</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem42</th>\n",
       "      <td>HRAD17(Rad24)- RFC complex</td>\n",
       "      <td>0.359540</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem40</th>\n",
       "      <td>RPA (replication factor A)</td>\n",
       "      <td>0.593273</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem401</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>0.499380</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem39</th>\n",
       "      <td>DNA polymerase epsilon</td>\n",
       "      <td>0.660657</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem36</th>\n",
       "      <td>UBL-conjugating enzymes</td>\n",
       "      <td>0.176977</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem35</th>\n",
       "      <td>RAD9-Hus1-Rad1 complex</td>\n",
       "      <td>0.359079</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem34</th>\n",
       "      <td>Rad17-Mec3- Ddc1 complex</td>\n",
       "      <td>0.446579</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem33</th>\n",
       "      <td>DNA replication</td>\n",
       "      <td>0.362578</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem2</th>\n",
       "      <td>S-CC phase</td>\n",
       "      <td>0.878101</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem29</th>\n",
       "      <td>MutL homologs</td>\n",
       "      <td>0.444021</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem290</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>0.173701</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem28</th>\n",
       "      <td>Ubiquitin- activating enzymes (E1)</td>\n",
       "      <td>0.425988</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem280</th>\n",
       "      <td>Ubiquitin response</td>\n",
       "      <td>0.186222</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem1</th>\n",
       "      <td>AP endonucleases</td>\n",
       "      <td>0.879606</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem194</th>\n",
       "      <td>Chromosome segregation</td>\n",
       "      <td>0.158264</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem18</th>\n",
       "      <td>Other MMR factors</td>\n",
       "      <td>0.447440</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem164</th>\n",
       "      <td>DNA glycosylases</td>\n",
       "      <td>0.171189</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem15</th>\n",
       "      <td>SHORT PATCH-BER factors</td>\n",
       "      <td>0.592860</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem157</th>\n",
       "      <td>Other single Ring- finger type E3</td>\n",
       "      <td>0.223494</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem156</th>\n",
       "      <td>Mismatch and loop recognition factors</td>\n",
       "      <td>0.787846</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem155</th>\n",
       "      <td>Ubiquitin- conjugating enzymes (E2)</td>\n",
       "      <td>0.423968</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem14</th>\n",
       "      <td>Ubiquitin-conjugating enzymes</td>\n",
       "      <td>0.908627</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem149</th>\n",
       "      <td>G1-S checkpoint</td>\n",
       "      <td>0.359365</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem134</th>\n",
       "      <td>p53 pathway</td>\n",
       "      <td>0.360663</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem12</th>\n",
       "      <td>G2-CC phase</td>\n",
       "      <td>0.592873</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem116</th>\n",
       "      <td>G2-M checkpoint</td>\n",
       "      <td>0.203809</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subsystem0</th>\n",
       "      <td>Direct Repair</td>\n",
       "      <td>0.601929</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               Term  Similarity  FDR\n",
       "Term                                                                \n",
       "subsystem37                              Ubiquitins    0.444618  0.0\n",
       "subsystem31                                    SUMO    0.444054  0.0\n",
       "subsystem10                                Alt-NHEJ    0.439860  0.0\n",
       "subsystem396                                    SSR    0.297021  0.0\n",
       "subsystem269                                    TLS    0.231794  0.0\n",
       "subsystem391                                   NHEJ    0.189705  0.0\n",
       "subsystem451                                    BER    0.183639  0.0\n",
       "subsystem289                                    MMR    0.182943  0.0\n",
       "subsystem456                                    DSR    0.160267  0.0\n",
       "subsystem272                                    NER    0.129968  0.0\n",
       "subsystem96              RFC (replication factor C)    0.256521  0.0\n",
       "subsystem95           UBL-specific proteases (ULPs)    0.201562  0.0\n",
       "subsystem8            FPC (fork protection complex)    0.881386  0.0\n",
       "subsystem805                         Repair pathway    0.619082  0.0\n",
       "subsystem7    MutS homologs specialized for meiosis    0.887586  0.0\n",
       "subsystem6                              MRN Complex    0.594000  0.0\n",
       "subsystem5                        damage in S phase    0.881386  0.0\n",
       "subsystem58                        Riddle syndrome!    0.440850  0.0\n",
       "subsystem52          Modulation of nucleotide pools    0.314578  0.0\n",
       "subsystem4            Direct Repair (not in humans)    0.891414  0.0\n",
       "subsystem49                         epistasis group    0.296026  0.0\n",
       "subsystem42              HRAD17(Rad24)- RFC complex    0.359540  0.0\n",
       "subsystem40              RPA (replication factor A)    0.593273  0.0\n",
       "subsystem401                     Associated process    0.499380  0.0\n",
       "subsystem39                  DNA polymerase epsilon    0.660657  0.0\n",
       "subsystem36                 UBL-conjugating enzymes    0.176977  0.0\n",
       "subsystem35                  RAD9-Hus1-Rad1 complex    0.359079  0.0\n",
       "subsystem34                Rad17-Mec3- Ddc1 complex    0.446579  0.0\n",
       "subsystem33                         DNA replication    0.362578  0.0\n",
       "subsystem2                               S-CC phase    0.878101  0.0\n",
       "subsystem29                           MutL homologs    0.444021  0.0\n",
       "subsystem290                     Checkpoint factors    0.173701  0.0\n",
       "subsystem28      Ubiquitin- activating enzymes (E1)    0.425988  0.0\n",
       "subsystem280                     Ubiquitin response    0.186222  0.0\n",
       "subsystem1                         AP endonucleases    0.879606  0.0\n",
       "subsystem194                 Chromosome segregation    0.158264  0.0\n",
       "subsystem18                       Other MMR factors    0.447440  0.0\n",
       "subsystem164                       DNA glycosylases    0.171189  0.0\n",
       "subsystem15                 SHORT PATCH-BER factors    0.592860  0.0\n",
       "subsystem157      Other single Ring- finger type E3    0.223494  0.0\n",
       "subsystem156  Mismatch and loop recognition factors    0.787846  0.0\n",
       "subsystem155    Ubiquitin- conjugating enzymes (E2)    0.423968  0.0\n",
       "subsystem14           Ubiquitin-conjugating enzymes    0.908627  0.0\n",
       "subsystem149                        G1-S checkpoint    0.359365  0.0\n",
       "subsystem134                            p53 pathway    0.360663  0.0\n",
       "subsystem12                             G2-CC phase    0.592873  0.0\n",
       "subsystem116                        G2-M checkpoint    0.203809  0.0\n",
       "subsystem0                            Direct Repair    0.601929  0.0"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont.align(ont2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:.ddot]",
   "language": "python",
   "name": "conda-env-.ddot-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
