{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import gzip\n",
    "import pandas as pd\n",
    "import networkx as nx\n",
    "import sys\n",
    "\n",
    "import ddot\n",
    "from ddot import Ontology"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "ndex_server = 'http://test.ndexbio.org' \n",
    "\n",
    "# Set the NDEx server and the user account (replace with your own account)\n",
    "ndex_user, ndex_pass = 'xiaoran.a.yan', 'a1l9i8f4e3'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download GO obo file\n",
    "r = requests.get('http://purl.obolibrary.org/obo/go/go-basic.obo')\n",
    "with open('go-basic.obo', 'wb') as f:\n",
    "    f.write(r.content)\n",
    "\n",
    "# Parse OBO file\n",
    "ddot.parse_obo('go-basic.obo', 'go.tab', 'goID_2_name.tab', 'goID_2_namespace.tab', 'goID_2_alt_id.tab')\n",
    "\n",
    "# Download gene-term annotations for human\n",
    "r = requests.get('http://geneontology.org/gene-associations/goa_human.gaf.gz')\n",
    "with open('goa_human.gaf.gz', 'wb') as f:\n",
    "    f.write(r.content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "hierarchy = pd.read_table('go.tab',\n",
    "                          sep='\\t',\n",
    "                          header=None,\n",
    "                          names=['Parent', 'Child', 'Relation', 'Namespace'])\n",
    "with gzip.open('goa_human.gaf.gz', 'rb') as f:\n",
    "    mapping = ddot.parse_gaf(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Dropping 213667 duplicate gene-term connections\n",
      "Unifying 3 roots into one super-root\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "19626 genes, 44412 terms, 287930 gene-term relations, 89326 term-term relations\n",
       "node_attributes: []\n",
       "edge_attributes: []"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "go_human = Ontology.from_table(\n",
    "    table=hierarchy,\n",
    "    parent='Parent',\n",
    "    child='Child',\n",
    "    mapping=mapping,\n",
    "    mapping_child='DB Object ID',\n",
    "    mapping_parent='GO ID',\n",
    "    add_root_name='GO:00SUPER',\n",
    "    ignore_orphan_terms=True)\n",
    "go_human.clear_node_attr()\n",
    "go_human.clear_edge_attr()\n",
    "go_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "collapse command: /home/yan30/.ddot/lib/python3.7/site-packages/ddot/alignOntology/collapseRedundantNodes /tmp/tmpwhvvee09\n",
      "19626 genes, 20153 terms, 228993 gene-term relations, 44286 term-term relations\n",
      "node_attributes: []\n",
      "edge_attributes: []\n"
     ]
    }
   ],
   "source": [
    "go_human = go_human.collapse_ontology(method='mhkramer')\n",
    "if 'GO:00SUPER' not in go_human.terms: go_human.add_root('GO:00SUPER', inplace=True)\n",
    "print(go_human)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "go_descriptions = pd.read_table('goID_2_name.tab',\n",
    "                                header=None,\n",
    "                                names=['Term', 'Term_Description'],\n",
    "                                index_col=0)\n",
    "go_human.update_node_attr(go_descriptions)\n",
    "\n",
    "go_branches = pd.read_table('goID_2_namespace.tab',\n",
    "                                header=None,\n",
    "                                names=['Term', 'Branch'],\n",
    "                                index_col=0)\n",
    "go_human.update_node_attr(go_branches)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:0000002</td>\n",
       "      <td>GO:0032042</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GO:0000003</td>\n",
       "      <td>GO:0019953</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GO:0000003</td>\n",
       "      <td>GO:0022414</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO:0000003</td>\n",
       "      <td>GO:0032504</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GO:0000003</td>\n",
       "      <td>GO:0071515</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GO:0000012</td>\n",
       "      <td>GO:1903516</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GO:0000012</td>\n",
       "      <td>GO:1903518</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GO:0000012</td>\n",
       "      <td>GO:1903824</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GO:0000014</td>\n",
       "      <td>GO:1990599</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0000019</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0010569</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0045191</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0045910</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0045911</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0032208</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0010845</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0098530</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0048239</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GO:0000019</td>\n",
       "      <td>GO:0045950</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GO:0000019</td>\n",
       "      <td>GO:0045951</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GO:0000019</td>\n",
       "      <td>GO:0032208</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GO:0000022</td>\n",
       "      <td>GO:0051256</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GO:0000022</td>\n",
       "      <td>GO:1902846</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GO:0000026</td>\n",
       "      <td>GO:0004377</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GO:0000026</td>\n",
       "      <td>GO:0052918</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GO:0000026</td>\n",
       "      <td>GO:0052926</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GO:0000027</td>\n",
       "      <td>GO:0034463</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GO:0000027</td>\n",
       "      <td>GO:1902775</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GO:0000028</td>\n",
       "      <td>GO:0034463</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GO:0000028</td>\n",
       "      <td>GO:0034462</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273249</th>\n",
       "      <td>GO:0006357</td>\n",
       "      <td>V9GYV3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273250</th>\n",
       "      <td>GO:0014066</td>\n",
       "      <td>V9GYV3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273251</th>\n",
       "      <td>GO:0051279</td>\n",
       "      <td>V9GYV3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273252</th>\n",
       "      <td>GO:0043122</td>\n",
       "      <td>V9GYV3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273253</th>\n",
       "      <td>GO:1901222</td>\n",
       "      <td>V9GYV3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273254</th>\n",
       "      <td>GO:2001294</td>\n",
       "      <td>V9GYW3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273255</th>\n",
       "      <td>GO:0050080</td>\n",
       "      <td>V9GYW3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273256</th>\n",
       "      <td>GO:0005782</td>\n",
       "      <td>V9GYW3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273257</th>\n",
       "      <td>GO:0046321</td>\n",
       "      <td>V9GYW3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273258</th>\n",
       "      <td>GO:0006085</td>\n",
       "      <td>V9GYW3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273259</th>\n",
       "      <td>GO:0005759</td>\n",
       "      <td>V9GYW3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273260</th>\n",
       "      <td>GO:0005730</td>\n",
       "      <td>V9GYY5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273261</th>\n",
       "      <td>GO:0003723</td>\n",
       "      <td>V9GYY5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273262</th>\n",
       "      <td>GO:0016021</td>\n",
       "      <td>V9GZ13</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273263</th>\n",
       "      <td>GO:0016021</td>\n",
       "      <td>W5XKT8</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273264</th>\n",
       "      <td>GO:0007342</td>\n",
       "      <td>W5XKT8</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273265</th>\n",
       "      <td>GO:0005515</td>\n",
       "      <td>W5XKT8</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273266</th>\n",
       "      <td>GO:0097169</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273267</th>\n",
       "      <td>GO:0002218</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273268</th>\n",
       "      <td>GO:0003690</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273269</th>\n",
       "      <td>GO:0005515</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273270</th>\n",
       "      <td>GO:0035458</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273271</th>\n",
       "      <td>GO:0098586</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273272</th>\n",
       "      <td>GO:0031333</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273273</th>\n",
       "      <td>GO:0050718</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273274</th>\n",
       "      <td>GO:0006954</td>\n",
       "      <td>W6CW81</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273275</th>\n",
       "      <td>GO:0016021</td>\n",
       "      <td>X5D2U9</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273276</th>\n",
       "      <td>GO:0006955</td>\n",
       "      <td>X5D2U9</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273277</th>\n",
       "      <td>GO:0019882</td>\n",
       "      <td>X5D2U9</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273278</th>\n",
       "      <td>GO:0042613</td>\n",
       "      <td>X5D2U9</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>273279 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Parent       Child      EdgeType\n",
       "0       GO:0000002  GO:0032042  Child-Parent\n",
       "1       GO:0000003  GO:0019953  Child-Parent\n",
       "2       GO:0000003  GO:0022414  Child-Parent\n",
       "3       GO:0000003  GO:0032504  Child-Parent\n",
       "4       GO:0000003  GO:0071515  Child-Parent\n",
       "5       GO:0000012  GO:1903516  Child-Parent\n",
       "6       GO:0000012  GO:1903518  Child-Parent\n",
       "7       GO:0000012  GO:1903824  Child-Parent\n",
       "8       GO:0000014  GO:1990599  Child-Parent\n",
       "9       GO:0000018  GO:0000019  Child-Parent\n",
       "10      GO:0000018  GO:0010569  Child-Parent\n",
       "11      GO:0000018  GO:0045191  Child-Parent\n",
       "12      GO:0000018  GO:0045910  Child-Parent\n",
       "13      GO:0000018  GO:0045911  Child-Parent\n",
       "14      GO:0000018  GO:0032208  Child-Parent\n",
       "15      GO:0000018  GO:0010845  Child-Parent\n",
       "16      GO:0000018  GO:0098530  Child-Parent\n",
       "17      GO:0000018  GO:0048239  Child-Parent\n",
       "18      GO:0000019  GO:0045950  Child-Parent\n",
       "19      GO:0000019  GO:0045951  Child-Parent\n",
       "20      GO:0000019  GO:0032208  Child-Parent\n",
       "21      GO:0000022  GO:0051256  Child-Parent\n",
       "22      GO:0000022  GO:1902846  Child-Parent\n",
       "23      GO:0000026  GO:0004377  Child-Parent\n",
       "24      GO:0000026  GO:0052918  Child-Parent\n",
       "25      GO:0000026  GO:0052926  Child-Parent\n",
       "26      GO:0000027  GO:0034463  Child-Parent\n",
       "27      GO:0000027  GO:1902775  Child-Parent\n",
       "28      GO:0000028  GO:0034463  Child-Parent\n",
       "29      GO:0000028  GO:0034462  Child-Parent\n",
       "...            ...         ...           ...\n",
       "273249  GO:0006357      V9GYV3     Gene-Term\n",
       "273250  GO:0014066      V9GYV3     Gene-Term\n",
       "273251  GO:0051279      V9GYV3     Gene-Term\n",
       "273252  GO:0043122      V9GYV3     Gene-Term\n",
       "273253  GO:1901222      V9GYV3     Gene-Term\n",
       "273254  GO:2001294      V9GYW3     Gene-Term\n",
       "273255  GO:0050080      V9GYW3     Gene-Term\n",
       "273256  GO:0005782      V9GYW3     Gene-Term\n",
       "273257  GO:0046321      V9GYW3     Gene-Term\n",
       "273258  GO:0006085      V9GYW3     Gene-Term\n",
       "273259  GO:0005759      V9GYW3     Gene-Term\n",
       "273260  GO:0005730      V9GYY5     Gene-Term\n",
       "273261  GO:0003723      V9GYY5     Gene-Term\n",
       "273262  GO:0016021      V9GZ13     Gene-Term\n",
       "273263  GO:0016021      W5XKT8     Gene-Term\n",
       "273264  GO:0007342      W5XKT8     Gene-Term\n",
       "273265  GO:0005515      W5XKT8     Gene-Term\n",
       "273266  GO:0097169      W6CW81     Gene-Term\n",
       "273267  GO:0002218      W6CW81     Gene-Term\n",
       "273268  GO:0003690      W6CW81     Gene-Term\n",
       "273269  GO:0005515      W6CW81     Gene-Term\n",
       "273270  GO:0035458      W6CW81     Gene-Term\n",
       "273271  GO:0098586      W6CW81     Gene-Term\n",
       "273272  GO:0031333      W6CW81     Gene-Term\n",
       "273273  GO:0050718      W6CW81     Gene-Term\n",
       "273274  GO:0006954      W6CW81     Gene-Term\n",
       "273275  GO:0016021      X5D2U9     Gene-Term\n",
       "273276  GO:0006955      X5D2U9     Gene-Term\n",
       "273277  GO:0019882      X5D2U9     Gene-Term\n",
       "273278  GO:0042613      X5D2U9     Gene-Term\n",
       "\n",
       "[273279 rows x 3 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "go_human.to_table('go_human.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "import mygene\n",
    "mg = mygene.MyGeneInfo()\n",
    "name = 'Human-specific Gene Ontology'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://test.ndexbio.org/v2/network/91bd87aa-bb70-11ea-a4d3-0660b7976219\n"
     ]
    }
   ],
   "source": [
    "go_human_uniprot = go_human.copy()\n",
    "\n",
    "# Write GO to file\n",
    "go_human_uniprot.to_table('collapsed_go.uniprot', clixo_format=True)\n",
    "go_human_uniprot.to_pickle('collapsed_go.uniprot.pkl')\n",
    "\n",
    "url, G = go_human_uniprot.to_ndex(name='%s, %s' % (name, 'UniProt'),\n",
    "                                  ndex_server=ndex_server,\n",
    "                                  ndex_user=ndex_user,\n",
    "                                  ndex_pass=ndex_pass,\n",
    "                                  layout=None,\n",
    "                                  visibility='PUBLIC')\n",
    "print(url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "querying 1-1000...done.\n",
      "querying 1001-2000...done.\n",
      "querying 2001-3000...done.\n",
      "querying 3001-4000...done.\n",
      "querying 4001-5000...done.\n",
      "querying 5001-6000...done.\n",
      "querying 6001-7000...done.\n",
      "querying 7001-8000...done.\n",
      "querying 8001-9000...done.\n",
      "querying 9001-10000...done.\n",
      "querying 10001-11000...done.\n",
      "querying 11001-12000...done.\n",
      "querying 12001-13000...done.\n",
      "querying 13001-14000...done.\n",
      "querying 14001-15000...done.\n",
      "querying 15001-16000...done.\n",
      "querying 16001-17000...done.\n",
      "querying 17001-18000...done.\n",
      "querying 18001-19000...done.\n",
      "querying 19001-19626...done.\n",
      "Finished.\n",
      "321 input query terms found dup hits:\n",
      "\t[('A0A075B6H7', 2), ('A0A075B6J1', 2), ('A0A075B6J2', 2), ('A0A075B6P5', 2), ('A0A075B6Q5', 2), ('A0\n",
      "731 input query terms found no hit:\n",
      "\t['A0A075B734', 'A0A075B767', 'A0A087WSY4', 'A0A087WUL8', 'A0A087WUU8', 'A0A087WX78', 'A0A087X1C1', '\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n"
     ]
    }
   ],
   "source": [
    "uniprot_2_symbol_df = mg.querymany(go_human.genes, scopes='uniprot', fields='symbol', species='human', as_dataframe=True)\n",
    "\n",
    "def f(x):\n",
    "    x = x['symbol']\n",
    "    if len(x)==1:\n",
    "        return x[0]\n",
    "    else:\n",
    "        return x.tolist()\n",
    "uniprot_2_symbol = uniprot_2_symbol_df.dropna(subset=['symbol']).groupby('query').apply(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19021 genes, 44412 terms, 285936 gene-term relations, 89326 term-term relations\n",
      "node_attributes: ['Term_Description', 'Branch']\n",
      "edge_attributes: []\n",
      "https://test.ndexbio.org/v2/network/bd0eb3cd-bb70-11ea-a4d3-0660b7976219\n"
     ]
    }
   ],
   "source": [
    "go_human_symbol = go_human.delete(to_delete=set(go_human.genes) - set(uniprot_2_symbol.keys()))\n",
    "go_human_symbol = go_human_symbol.rename(genes=uniprot_2_symbol.to_dict())\n",
    "print(go_human_symbol)\n",
    "\n",
    "# Write GO to file\n",
    "go_human_symbol.to_table('collapsed_go.symbol', clixo_format=True)\n",
    "go_human_symbol.to_pickle('collapsed_go.symbol.pkl')\n",
    "\n",
    "url, G = go_human_symbol.to_ndex(name='%s, %s' % (name, 'Symbol'),\n",
    "                                 ndex_server=ndex_server,\n",
    "                                 ndex_user=ndex_user,\n",
    "                                 ndex_pass=ndex_pass,\n",
    "                                 layout=None,\n",
    "                                 visibility='PUBLIC')\n",
    "print(url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:0000002</td>\n",
       "      <td>GO:0032042</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GO:0000003</td>\n",
       "      <td>GO:0019953</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GO:0000003</td>\n",
       "      <td>GO:0022414</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO:0000003</td>\n",
       "      <td>GO:0032504</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GO:0000003</td>\n",
       "      <td>GO:0071515</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GO:0000012</td>\n",
       "      <td>GO:1903516</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GO:0000012</td>\n",
       "      <td>GO:1903518</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GO:0000012</td>\n",
       "      <td>GO:1903824</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GO:0000014</td>\n",
       "      <td>GO:1990599</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0000019</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0010569</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0045191</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0045910</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0045911</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0032208</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0010845</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0098530</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GO:0000018</td>\n",
       "      <td>GO:0048239</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GO:0000019</td>\n",
       "      <td>GO:0045950</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GO:0000019</td>\n",
       "      <td>GO:0045951</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GO:0000019</td>\n",
       "      <td>GO:0032208</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GO:0000022</td>\n",
       "      <td>GO:0051256</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GO:0000022</td>\n",
       "      <td>GO:1902846</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GO:0000026</td>\n",
       "      <td>GO:0004377</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GO:0000026</td>\n",
       "      <td>GO:0052918</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GO:0000026</td>\n",
       "      <td>GO:0052926</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GO:0000027</td>\n",
       "      <td>GO:0034463</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GO:0000027</td>\n",
       "      <td>GO:1902775</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GO:0000028</td>\n",
       "      <td>GO:0034463</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GO:0000028</td>\n",
       "      <td>GO:0034462</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271280</th>\n",
       "      <td>GO:0005634</td>\n",
       "      <td>AC012531.3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271281</th>\n",
       "      <td>GO:0009952</td>\n",
       "      <td>AC012531.3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271282</th>\n",
       "      <td>GO:0033613</td>\n",
       "      <td>AC012531.3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271283</th>\n",
       "      <td>GO:0048704</td>\n",
       "      <td>AC012531.3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271284</th>\n",
       "      <td>GO:0000977</td>\n",
       "      <td>AC012531.3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271285</th>\n",
       "      <td>GO:0001228</td>\n",
       "      <td>AC012531.3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271286</th>\n",
       "      <td>GO:0019815</td>\n",
       "      <td>AL121845.2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271287</th>\n",
       "      <td>GO:0019901</td>\n",
       "      <td>AL121845.2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271288</th>\n",
       "      <td>GO:0050853</td>\n",
       "      <td>AL121845.2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271289</th>\n",
       "      <td>GO:0043405</td>\n",
       "      <td>AL121845.2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271290</th>\n",
       "      <td>GO:0006357</td>\n",
       "      <td>AL121845.2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271291</th>\n",
       "      <td>GO:0014066</td>\n",
       "      <td>AL121845.2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271292</th>\n",
       "      <td>GO:0051279</td>\n",
       "      <td>AL121845.2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271293</th>\n",
       "      <td>GO:0043122</td>\n",
       "      <td>AL121845.2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271294</th>\n",
       "      <td>GO:1901222</td>\n",
       "      <td>AL121845.2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271295</th>\n",
       "      <td>GO:0005730</td>\n",
       "      <td>Z83844.1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271296</th>\n",
       "      <td>GO:0003723</td>\n",
       "      <td>Z83844.1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271297</th>\n",
       "      <td>GO:0016021</td>\n",
       "      <td>AL034430.1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271298</th>\n",
       "      <td>GO:0016021</td>\n",
       "      <td>SPACA6</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271299</th>\n",
       "      <td>GO:0007342</td>\n",
       "      <td>SPACA6</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271300</th>\n",
       "      <td>GO:0005515</td>\n",
       "      <td>SPACA6</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271301</th>\n",
       "      <td>GO:0097169</td>\n",
       "      <td>PYDC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271302</th>\n",
       "      <td>GO:0002218</td>\n",
       "      <td>PYDC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271303</th>\n",
       "      <td>GO:0003690</td>\n",
       "      <td>PYDC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271304</th>\n",
       "      <td>GO:0005515</td>\n",
       "      <td>PYDC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271305</th>\n",
       "      <td>GO:0035458</td>\n",
       "      <td>PYDC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271306</th>\n",
       "      <td>GO:0098586</td>\n",
       "      <td>PYDC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271307</th>\n",
       "      <td>GO:0031333</td>\n",
       "      <td>PYDC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271308</th>\n",
       "      <td>GO:0050718</td>\n",
       "      <td>PYDC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271309</th>\n",
       "      <td>GO:0006954</td>\n",
       "      <td>PYDC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>271310 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Parent       Child      EdgeType\n",
       "0       GO:0000002  GO:0032042  Child-Parent\n",
       "1       GO:0000003  GO:0019953  Child-Parent\n",
       "2       GO:0000003  GO:0022414  Child-Parent\n",
       "3       GO:0000003  GO:0032504  Child-Parent\n",
       "4       GO:0000003  GO:0071515  Child-Parent\n",
       "5       GO:0000012  GO:1903516  Child-Parent\n",
       "6       GO:0000012  GO:1903518  Child-Parent\n",
       "7       GO:0000012  GO:1903824  Child-Parent\n",
       "8       GO:0000014  GO:1990599  Child-Parent\n",
       "9       GO:0000018  GO:0000019  Child-Parent\n",
       "10      GO:0000018  GO:0010569  Child-Parent\n",
       "11      GO:0000018  GO:0045191  Child-Parent\n",
       "12      GO:0000018  GO:0045910  Child-Parent\n",
       "13      GO:0000018  GO:0045911  Child-Parent\n",
       "14      GO:0000018  GO:0032208  Child-Parent\n",
       "15      GO:0000018  GO:0010845  Child-Parent\n",
       "16      GO:0000018  GO:0098530  Child-Parent\n",
       "17      GO:0000018  GO:0048239  Child-Parent\n",
       "18      GO:0000019  GO:0045950  Child-Parent\n",
       "19      GO:0000019  GO:0045951  Child-Parent\n",
       "20      GO:0000019  GO:0032208  Child-Parent\n",
       "21      GO:0000022  GO:0051256  Child-Parent\n",
       "22      GO:0000022  GO:1902846  Child-Parent\n",
       "23      GO:0000026  GO:0004377  Child-Parent\n",
       "24      GO:0000026  GO:0052918  Child-Parent\n",
       "25      GO:0000026  GO:0052926  Child-Parent\n",
       "26      GO:0000027  GO:0034463  Child-Parent\n",
       "27      GO:0000027  GO:1902775  Child-Parent\n",
       "28      GO:0000028  GO:0034463  Child-Parent\n",
       "29      GO:0000028  GO:0034462  Child-Parent\n",
       "...            ...         ...           ...\n",
       "271280  GO:0005634  AC012531.3     Gene-Term\n",
       "271281  GO:0009952  AC012531.3     Gene-Term\n",
       "271282  GO:0033613  AC012531.3     Gene-Term\n",
       "271283  GO:0048704  AC012531.3     Gene-Term\n",
       "271284  GO:0000977  AC012531.3     Gene-Term\n",
       "271285  GO:0001228  AC012531.3     Gene-Term\n",
       "271286  GO:0019815  AL121845.2     Gene-Term\n",
       "271287  GO:0019901  AL121845.2     Gene-Term\n",
       "271288  GO:0050853  AL121845.2     Gene-Term\n",
       "271289  GO:0043405  AL121845.2     Gene-Term\n",
       "271290  GO:0006357  AL121845.2     Gene-Term\n",
       "271291  GO:0014066  AL121845.2     Gene-Term\n",
       "271292  GO:0051279  AL121845.2     Gene-Term\n",
       "271293  GO:0043122  AL121845.2     Gene-Term\n",
       "271294  GO:1901222  AL121845.2     Gene-Term\n",
       "271295  GO:0005730    Z83844.1     Gene-Term\n",
       "271296  GO:0003723    Z83844.1     Gene-Term\n",
       "271297  GO:0016021  AL034430.1     Gene-Term\n",
       "271298  GO:0016021      SPACA6     Gene-Term\n",
       "271299  GO:0007342      SPACA6     Gene-Term\n",
       "271300  GO:0005515      SPACA6     Gene-Term\n",
       "271301  GO:0097169       PYDC5     Gene-Term\n",
       "271302  GO:0002218       PYDC5     Gene-Term\n",
       "271303  GO:0003690       PYDC5     Gene-Term\n",
       "271304  GO:0005515       PYDC5     Gene-Term\n",
       "271305  GO:0035458       PYDC5     Gene-Term\n",
       "271306  GO:0098586       PYDC5     Gene-Term\n",
       "271307  GO:0031333       PYDC5     Gene-Term\n",
       "271308  GO:0050718       PYDC5     Gene-Term\n",
       "271309  GO:0006954       PYDC5     Gene-Term\n",
       "\n",
       "[271310 rows x 3 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "go_human_symbol.to_table('go_human.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"ALKBH3\" in go_human_symbol.genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "306 genes, 394 terms, 357 gene-term relations, 470 term-term relations\n",
      "node_attributes: []\n",
      "edge_attributes: []\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>subsystem100</td>\n",
       "      <td>subsystem85</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>subsystem100</td>\n",
       "      <td>subsystem88</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>subsystem101</td>\n",
       "      <td>subsystem86</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>subsystem101</td>\n",
       "      <td>subsystem94</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>subsystem102</td>\n",
       "      <td>subsystem87</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>subsystem102</td>\n",
       "      <td>subsystem93</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>subsystem104</td>\n",
       "      <td>subsystem3</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>subsystem104</td>\n",
       "      <td>subsystem6</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>subsystem108</td>\n",
       "      <td>subsystem41</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>subsystem109</td>\n",
       "      <td>subsystem95</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>subsystem109</td>\n",
       "      <td>subsystem101</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>subsystem111</td>\n",
       "      <td>subsystem30</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>subsystem111</td>\n",
       "      <td>subsystem41</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>subsystem113</td>\n",
       "      <td>subsystem41</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>subsystem113</td>\n",
       "      <td>subsystem69</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>subsystem116</td>\n",
       "      <td>subsystem23</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>subsystem119</td>\n",
       "      <td>subsystem42</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>subsystem120</td>\n",
       "      <td>subsystem105</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>subsystem120</td>\n",
       "      <td>subsystem107</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>subsystem120</td>\n",
       "      <td>subsystem110</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>subsystem121</td>\n",
       "      <td>subsystem106</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>subsystem121</td>\n",
       "      <td>subsystem108</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>subsystem126</td>\n",
       "      <td>subsystem67</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>subsystem127</td>\n",
       "      <td>subsystem68</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>subsystem127</td>\n",
       "      <td>subsystem90</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>subsystem128</td>\n",
       "      <td>subsystem68</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>subsystem129</td>\n",
       "      <td>subsystem39</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>subsystem129</td>\n",
       "      <td>subsystem68</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>subsystem131</td>\n",
       "      <td>subsystem4</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>subsystem132</td>\n",
       "      <td>subsystem96</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>797</th>\n",
       "      <td>subsystem179</td>\n",
       "      <td>TUBGCP4</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>798</th>\n",
       "      <td>subsystem173</td>\n",
       "      <td>TUBGCP4</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>799</th>\n",
       "      <td>subsystem88</td>\n",
       "      <td>UBE2A</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>800</th>\n",
       "      <td>subsystem13</td>\n",
       "      <td>UBE2B</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>801</th>\n",
       "      <td>subsystem210</td>\n",
       "      <td>UBE2I</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>802</th>\n",
       "      <td>subsystem97</td>\n",
       "      <td>UBE2N</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>803</th>\n",
       "      <td>subsystem351</td>\n",
       "      <td>UBE2T</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>subsystem128</td>\n",
       "      <td>UBE2V2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>subsystem259</td>\n",
       "      <td>UBR7</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>subsystem114</td>\n",
       "      <td>UIMC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>subsystem136</td>\n",
       "      <td>UNG</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>subsystem359</td>\n",
       "      <td>USP1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>subsystem162</td>\n",
       "      <td>USP42</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>subsystem164</td>\n",
       "      <td>USP7</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>subsystem28</td>\n",
       "      <td>UVSSA</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>812</th>\n",
       "      <td>subsystem5</td>\n",
       "      <td>VRK1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>813</th>\n",
       "      <td>subsystem351</td>\n",
       "      <td>WDR4</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>814</th>\n",
       "      <td>subsystem16</td>\n",
       "      <td>WDR48</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>815</th>\n",
       "      <td>subsystem219</td>\n",
       "      <td>WRAP53</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>816</th>\n",
       "      <td>subsystem364</td>\n",
       "      <td>WRN</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>817</th>\n",
       "      <td>subsystem97</td>\n",
       "      <td>XAB2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>818</th>\n",
       "      <td>subsystem210</td>\n",
       "      <td>XPA</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>subsystem139</td>\n",
       "      <td>XPC</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820</th>\n",
       "      <td>subsystem291</td>\n",
       "      <td>XRCC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>821</th>\n",
       "      <td>subsystem381</td>\n",
       "      <td>XRCC2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>822</th>\n",
       "      <td>subsystem380</td>\n",
       "      <td>XRCC2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>823</th>\n",
       "      <td>subsystem341</td>\n",
       "      <td>XRCC3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>824</th>\n",
       "      <td>subsystem105</td>\n",
       "      <td>XRCC4</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>825</th>\n",
       "      <td>subsystem349</td>\n",
       "      <td>XRCC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826</th>\n",
       "      <td>subsystem198</td>\n",
       "      <td>XRCC6</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>827 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Parent         Child      EdgeType\n",
       "0    subsystem100   subsystem85  Child-Parent\n",
       "1    subsystem100   subsystem88  Child-Parent\n",
       "2    subsystem101   subsystem86  Child-Parent\n",
       "3    subsystem101   subsystem94  Child-Parent\n",
       "4    subsystem102   subsystem87  Child-Parent\n",
       "5    subsystem102   subsystem93  Child-Parent\n",
       "6    subsystem104    subsystem3  Child-Parent\n",
       "7    subsystem104    subsystem6  Child-Parent\n",
       "8    subsystem108   subsystem41  Child-Parent\n",
       "9    subsystem109   subsystem95  Child-Parent\n",
       "10   subsystem109  subsystem101  Child-Parent\n",
       "11   subsystem111   subsystem30  Child-Parent\n",
       "12   subsystem111   subsystem41  Child-Parent\n",
       "13   subsystem113   subsystem41  Child-Parent\n",
       "14   subsystem113   subsystem69  Child-Parent\n",
       "15   subsystem116   subsystem23  Child-Parent\n",
       "16   subsystem119   subsystem42  Child-Parent\n",
       "17   subsystem120  subsystem105  Child-Parent\n",
       "18   subsystem120  subsystem107  Child-Parent\n",
       "19   subsystem120  subsystem110  Child-Parent\n",
       "20   subsystem121  subsystem106  Child-Parent\n",
       "21   subsystem121  subsystem108  Child-Parent\n",
       "22   subsystem126   subsystem67  Child-Parent\n",
       "23   subsystem127   subsystem68  Child-Parent\n",
       "24   subsystem127   subsystem90  Child-Parent\n",
       "25   subsystem128   subsystem68  Child-Parent\n",
       "26   subsystem129   subsystem39  Child-Parent\n",
       "27   subsystem129   subsystem68  Child-Parent\n",
       "28   subsystem131    subsystem4  Child-Parent\n",
       "29   subsystem132   subsystem96  Child-Parent\n",
       "..            ...           ...           ...\n",
       "797  subsystem179       TUBGCP4     Gene-Term\n",
       "798  subsystem173       TUBGCP4     Gene-Term\n",
       "799   subsystem88         UBE2A     Gene-Term\n",
       "800   subsystem13         UBE2B     Gene-Term\n",
       "801  subsystem210         UBE2I     Gene-Term\n",
       "802   subsystem97         UBE2N     Gene-Term\n",
       "803  subsystem351         UBE2T     Gene-Term\n",
       "804  subsystem128        UBE2V2     Gene-Term\n",
       "805  subsystem259          UBR7     Gene-Term\n",
       "806  subsystem114         UIMC1     Gene-Term\n",
       "807  subsystem136           UNG     Gene-Term\n",
       "808  subsystem359          USP1     Gene-Term\n",
       "809  subsystem162         USP42     Gene-Term\n",
       "810  subsystem164          USP7     Gene-Term\n",
       "811   subsystem28         UVSSA     Gene-Term\n",
       "812    subsystem5          VRK1     Gene-Term\n",
       "813  subsystem351          WDR4     Gene-Term\n",
       "814   subsystem16         WDR48     Gene-Term\n",
       "815  subsystem219        WRAP53     Gene-Term\n",
       "816  subsystem364           WRN     Gene-Term\n",
       "817   subsystem97          XAB2     Gene-Term\n",
       "818  subsystem210           XPA     Gene-Term\n",
       "819  subsystem139           XPC     Gene-Term\n",
       "820  subsystem291         XRCC1     Gene-Term\n",
       "821  subsystem381         XRCC2     Gene-Term\n",
       "822  subsystem380         XRCC2     Gene-Term\n",
       "823  subsystem341         XRCC3     Gene-Term\n",
       "824  subsystem105         XRCC4     Gene-Term\n",
       "825  subsystem349         XRCC5     Gene-Term\n",
       "826  subsystem198         XRCC6     Gene-Term\n",
       "\n",
       "[827 rows x 3 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pickle\n",
    "\n",
    "with open ('hierarchyQnew0.save', 'rb') as fp:\n",
    "    itemlist = pickle.load(fp)\n",
    "\n",
    "hierarchy  = [x for x in itemlist if \"subsystem\" in str(x[0])]\n",
    "mapping = [x for x in itemlist if \"subsystem\" not in str(x[0])]\n",
    "\n",
    "import ddot\n",
    "from ddot import Ontology\n",
    "# Construct ontology\n",
    "ont = Ontology(hierarchy, mapping)\n",
    "\n",
    "# Prints a summary of the ontology's structure\n",
    "print(ont)\n",
    "ont.to_table('toy_ontology2.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ACD',\n",
       " 'ADNP',\n",
       " 'ALKBH2',\n",
       " 'ALKBH3',\n",
       " 'APEX1',\n",
       " 'APEX2',\n",
       " 'APLF',\n",
       " 'APTX',\n",
       " 'ATAD5',\n",
       " 'ATM',\n",
       " 'ATR',\n",
       " 'ATRIP',\n",
       " 'ATRX',\n",
       " 'BARD1',\n",
       " 'BCCIP',\n",
       " 'BLM',\n",
       " 'BRAT1',\n",
       " 'BRCA1',\n",
       " 'BRCA2',\n",
       " 'BRCC3',\n",
       " 'BRIP1',\n",
       " 'CBX3',\n",
       " 'CCNA2',\n",
       " 'CCNH',\n",
       " 'CDC45',\n",
       " 'CDC7',\n",
       " 'CDK2',\n",
       " 'CDK7',\n",
       " 'CENPK',\n",
       " 'CEP152',\n",
       " 'CETN2',\n",
       " 'CETN3',\n",
       " 'CHAF1A',\n",
       " 'CHEK1',\n",
       " 'CHEK2',\n",
       " 'CHTF18',\n",
       " 'CKS2',\n",
       " 'CLK2',\n",
       " 'CLSPN',\n",
       " 'DBF4B',\n",
       " 'DCLRE1A',\n",
       " 'DCLRE1B',\n",
       " 'DCLRE1C',\n",
       " 'DDB1',\n",
       " 'DDB2',\n",
       " 'DHX9',\n",
       " 'DMC1',\n",
       " 'DNA2',\n",
       " 'DRG1',\n",
       " 'DSCC1',\n",
       " 'DTL',\n",
       " 'DUT',\n",
       " 'EME1',\n",
       " 'EME2',\n",
       " 'ENDOV',\n",
       " 'ERCC1',\n",
       " 'ERCC2',\n",
       " 'ERCC3',\n",
       " 'ERCC4',\n",
       " 'ERCC5',\n",
       " 'ERCC6',\n",
       " 'ERCC8',\n",
       " 'EXO1',\n",
       " 'FAAP100',\n",
       " 'FAAP20',\n",
       " 'FAAP24',\n",
       " 'FAN1',\n",
       " 'FANCA',\n",
       " 'FANCB',\n",
       " 'FANCC',\n",
       " 'FANCD2',\n",
       " 'FANCE',\n",
       " 'FANCF',\n",
       " 'FANCG',\n",
       " 'FANCI',\n",
       " 'FANCL',\n",
       " 'FANCM',\n",
       " 'FEN1',\n",
       " 'GADD45A',\n",
       " 'GADD45G',\n",
       " 'GEN1',\n",
       " 'GGCT',\n",
       " 'GINS1',\n",
       " 'GINS2',\n",
       " 'GINS3',\n",
       " 'GINS4',\n",
       " 'GMPS',\n",
       " 'GTF2H1',\n",
       " 'GTF2H2',\n",
       " 'GTF2H3',\n",
       " 'GTF2H4',\n",
       " 'GTF2H5',\n",
       " 'HELQ',\n",
       " 'HLTF',\n",
       " 'HMGB1',\n",
       " 'HMGB2',\n",
       " 'HNRNPR',\n",
       " 'HNRNPUL1',\n",
       " 'HUS1',\n",
       " 'HUS1B',\n",
       " 'KIN',\n",
       " 'LIG1',\n",
       " 'LIG3',\n",
       " 'LIG4',\n",
       " 'MAD2L2',\n",
       " 'MBD4',\n",
       " 'MCM10',\n",
       " 'MCM2',\n",
       " 'MCM3',\n",
       " 'MCM4',\n",
       " 'MCM5',\n",
       " 'MCM6',\n",
       " 'MCM7',\n",
       " 'MCM8',\n",
       " 'MCMBP',\n",
       " 'MDC1',\n",
       " 'MGMT',\n",
       " 'MLH1',\n",
       " 'MLH3',\n",
       " 'MMS19',\n",
       " 'MMS22L',\n",
       " 'MNAT1',\n",
       " 'MPG',\n",
       " 'MPLKIP',\n",
       " 'MRE11',\n",
       " 'MSH2',\n",
       " 'MSH3',\n",
       " 'MSH4',\n",
       " 'MSH5',\n",
       " 'MSH6',\n",
       " 'MTBP',\n",
       " 'MUS81',\n",
       " 'MUTYH',\n",
       " 'NABP2',\n",
       " 'NARFL',\n",
       " 'NBN',\n",
       " 'NCAPD2',\n",
       " 'NCAPD3',\n",
       " 'NCAPG',\n",
       " 'NCAPG2',\n",
       " 'NCAPH',\n",
       " 'NCAPH2',\n",
       " 'NEIL1',\n",
       " 'NEIL2',\n",
       " 'NEIL3',\n",
       " 'NHEJ1',\n",
       " 'NTHL1',\n",
       " 'NUDT1',\n",
       " 'NUP188',\n",
       " 'OGG1',\n",
       " 'ORC1',\n",
       " 'ORC2',\n",
       " 'ORC5',\n",
       " 'ORC6',\n",
       " 'PALB2',\n",
       " 'PARP1',\n",
       " 'PARP2',\n",
       " 'PARP3',\n",
       " 'PARP4',\n",
       " 'PAXIP1',\n",
       " 'PCNA',\n",
       " 'PDS5A',\n",
       " 'PDS5B',\n",
       " 'PER1',\n",
       " 'PMS1',\n",
       " 'PMS2',\n",
       " 'PNKP',\n",
       " 'POLA1',\n",
       " 'POLA2',\n",
       " 'POLB',\n",
       " 'POLD1',\n",
       " 'POLD2',\n",
       " 'POLD3',\n",
       " 'POLD4',\n",
       " 'POLE',\n",
       " 'POLE2',\n",
       " 'POLE3',\n",
       " 'POLG',\n",
       " 'POLH',\n",
       " 'POLI',\n",
       " 'POLK',\n",
       " 'POLL',\n",
       " 'POLM',\n",
       " 'POLN',\n",
       " 'POLQ',\n",
       " 'POLR2D',\n",
       " 'POLR2G',\n",
       " 'POT1',\n",
       " 'PPAT',\n",
       " 'PRIM1',\n",
       " 'PRIM2',\n",
       " 'PRKDC',\n",
       " 'PRPF19',\n",
       " 'RAD1',\n",
       " 'RAD17',\n",
       " 'RAD18',\n",
       " 'RAD23A',\n",
       " 'RAD23B',\n",
       " 'RAD50',\n",
       " 'RAD51',\n",
       " 'RAD51AP1',\n",
       " 'RAD51B',\n",
       " 'RAD51C',\n",
       " 'RAD51D',\n",
       " 'RAD52',\n",
       " 'RAD54B',\n",
       " 'RAD54L',\n",
       " 'RAD9A',\n",
       " 'RBBP8',\n",
       " 'RDM1',\n",
       " 'RECQL',\n",
       " 'RECQL4',\n",
       " 'RECQL5',\n",
       " 'REV1',\n",
       " 'REV3L',\n",
       " 'RFC1',\n",
       " 'RFC2',\n",
       " 'RFC3',\n",
       " 'RFC4',\n",
       " 'RFC5',\n",
       " 'RFWD3',\n",
       " 'RIF1',\n",
       " 'RMI1',\n",
       " 'RMI2',\n",
       " 'RNF168',\n",
       " 'RNF4',\n",
       " 'RNF8',\n",
       " 'RPA1',\n",
       " 'RPA2',\n",
       " 'RPA3',\n",
       " 'RPA4',\n",
       " 'RRM2B',\n",
       " 'RRP9',\n",
       " 'SEM1',\n",
       " 'SENP1',\n",
       " 'SETMAR',\n",
       " 'SHPRH',\n",
       " 'SLF1',\n",
       " 'SLX1A',\n",
       " 'SLX1B',\n",
       " 'SLX4',\n",
       " 'SMARCA5',\n",
       " 'SMARCAD1',\n",
       " 'SMC1A',\n",
       " 'SMC2',\n",
       " 'SMC4',\n",
       " 'SMC6',\n",
       " 'SMUG1',\n",
       " 'SPATA5',\n",
       " 'SPATA5L1',\n",
       " 'SPO11',\n",
       " 'SPRTN',\n",
       " 'SSRP1',\n",
       " 'SUMO2',\n",
       " 'SUPT16H',\n",
       " 'TDG',\n",
       " 'TDP1',\n",
       " 'TDP2',\n",
       " 'TELO2',\n",
       " 'TERF2',\n",
       " 'TICRR',\n",
       " 'TIMELESS',\n",
       " 'TIPIN',\n",
       " 'TONSL',\n",
       " 'TOP1',\n",
       " 'TOP2A',\n",
       " 'TOP2B',\n",
       " 'TOP3A',\n",
       " 'TOPBP1',\n",
       " 'TP53',\n",
       " 'TP53BP1',\n",
       " 'TRA2B',\n",
       " 'TREX1',\n",
       " 'TREX2',\n",
       " 'TTF1',\n",
       " 'TUBGCP4',\n",
       " 'UBE2A',\n",
       " 'UBE2B',\n",
       " 'UBE2I',\n",
       " 'UBE2N',\n",
       " 'UBE2T',\n",
       " 'UBE2V2',\n",
       " 'UBR7',\n",
       " 'UIMC1',\n",
       " 'UNG',\n",
       " 'USP1',\n",
       " 'USP42',\n",
       " 'USP7',\n",
       " 'UVSSA',\n",
       " 'VRK1',\n",
       " 'WDR4',\n",
       " 'WDR48',\n",
       " 'WRAP53',\n",
       " 'WRN',\n",
       " 'XAB2',\n",
       " 'XPA',\n",
       " 'XPC',\n",
       " 'XRCC1',\n",
       " 'XRCC2',\n",
       " 'XRCC3',\n",
       " 'XRCC4',\n",
       " 'XRCC5',\n",
       " 'XRCC6']"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matchers = ont.genes\n",
    "matching = [s for s in matchers if any(xs in s for xs in go_human_symbol.genes)]\n",
    "matching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "303"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(matching)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19021 genes, 44412 terms, 285936 gene-term relations, 89326 term-term relations\n",
       "node_attributes: ['Term_Description', 'Branch']\n",
       "edge_attributes: []"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "go_human_symbol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "437 genes, 74 terms, 533 gene-term relations, 74 term-term relations\n",
      "node_attributes: []\n",
      "edge_attributes: [2]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>Child</th>\n",
       "      <th>EdgeType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Alternative mechanism</td>\n",
       "      <td>MRN Complex</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Chromatin remodelling</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Chromosome segregation</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Modulation of nucleotide pools</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>p53 pathway</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Telomere maintenance</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>TLS</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Topisomerase damage reversal</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Associated process</td>\n",
       "      <td>Ubiquitin response</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>BER</td>\n",
       "      <td>AP endonucleases</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>BER</td>\n",
       "      <td>DNA glycosylases</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>BER</td>\n",
       "      <td>LONG PATCH-BER factors</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>BER</td>\n",
       "      <td>Other BER factors</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>BER</td>\n",
       "      <td>SHORT PATCH-BER factors</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>damage in S phase</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>FPC (fork protection complex)</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>G1-CC phase</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>G1-S checkpoint</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>G2-CC phase</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>G2-M checkpoint</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>HRAD17(Rad24)- RFC complex</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>Rad17-Mec3- Ddc1 complex</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>RAD9-Hus1-Rad1 complex</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Checkpoint factors</td>\n",
       "      <td>S-CC phase</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>DSR</td>\n",
       "      <td>Alt-NHEJ</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>DSR</td>\n",
       "      <td>FA (Fanconi anemia pathway)</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>DSR</td>\n",
       "      <td>HR (Homologous Recombination)</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>DSR</td>\n",
       "      <td>NHEJ</td>\n",
       "      <td>Child-Parent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>HR (Homologous Recombination)</td>\n",
       "      <td>UIMC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>DNA glycosylases</td>\n",
       "      <td>UNG</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>UPF1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>FA (Fanconi anemia pathway)</td>\n",
       "      <td>USP1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>TLS</td>\n",
       "      <td>USP1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>UVRAG</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>TCR (Transcription coupled repair)</td>\n",
       "      <td>UVSSA</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>584</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>VCP</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>FA (Fanconi anemia pathway)</td>\n",
       "      <td>WDR48</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>TLS</td>\n",
       "      <td>WDR48</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>G1-CC phase</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>G2-CC phase</td>\n",
       "      <td>WEE1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>Telomere maintenance</td>\n",
       "      <td>WRAP53</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>HR (Homologous Recombination)</td>\n",
       "      <td>WRN</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>591</th>\n",
       "      <td>LONG PATCH-BER factors</td>\n",
       "      <td>WRN</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>592</th>\n",
       "      <td>TCR (Transcription coupled repair)</td>\n",
       "      <td>XAB2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>593</th>\n",
       "      <td>TCR (Transcription coupled repair)</td>\n",
       "      <td>XPA</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>594</th>\n",
       "      <td>GGR (Global genome repair)</td>\n",
       "      <td>XPC</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>595</th>\n",
       "      <td>TCR (Transcription coupled repair)</td>\n",
       "      <td>XRCC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>Alt-NHEJ</td>\n",
       "      <td>XRCC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>597</th>\n",
       "      <td>SHORT PATCH-BER factors</td>\n",
       "      <td>XRCC1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>598</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC2</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC4</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>601</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC5</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>NHEJ</td>\n",
       "      <td>XRCC6</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>603</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>XRCC6BP1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>604</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>YBX1</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>605</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>ZRANB3</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>606</th>\n",
       "      <td>Genes with probable DDR role</td>\n",
       "      <td>ZSWIM7</td>\n",
       "      <td>Gene-Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>607 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Parent                           Child  \\\n",
       "0                 Alternative mechanism                     MRN Complex   \n",
       "1                    Associated process              Checkpoint factors   \n",
       "2                    Associated process           Chromatin remodelling   \n",
       "3                    Associated process          Chromosome segregation   \n",
       "4                    Associated process                 DNA replication   \n",
       "5                    Associated process  Modulation of nucleotide pools   \n",
       "6                    Associated process                     p53 pathway   \n",
       "7                    Associated process            Telomere maintenance   \n",
       "8                    Associated process                             TLS   \n",
       "9                    Associated process    Topisomerase damage reversal   \n",
       "10                   Associated process              Ubiquitin response   \n",
       "11                                  BER                AP endonucleases   \n",
       "12                                  BER                DNA glycosylases   \n",
       "13                                  BER          LONG PATCH-BER factors   \n",
       "14                                  BER               Other BER factors   \n",
       "15                                  BER         SHORT PATCH-BER factors   \n",
       "16                   Checkpoint factors               damage in S phase   \n",
       "17                   Checkpoint factors   FPC (fork protection complex)   \n",
       "18                   Checkpoint factors                     G1-CC phase   \n",
       "19                   Checkpoint factors                 G1-S checkpoint   \n",
       "20                   Checkpoint factors                     G2-CC phase   \n",
       "21                   Checkpoint factors                 G2-M checkpoint   \n",
       "22                   Checkpoint factors      HRAD17(Rad24)- RFC complex   \n",
       "23                   Checkpoint factors        Rad17-Mec3- Ddc1 complex   \n",
       "24                   Checkpoint factors          RAD9-Hus1-Rad1 complex   \n",
       "25                   Checkpoint factors                      S-CC phase   \n",
       "26                                  DSR                        Alt-NHEJ   \n",
       "27                                  DSR     FA (Fanconi anemia pathway)   \n",
       "28                                  DSR   HR (Homologous Recombination)   \n",
       "29                                  DSR                            NHEJ   \n",
       "..                                  ...                             ...   \n",
       "577       HR (Homologous Recombination)                           UIMC1   \n",
       "578                    DNA glycosylases                             UNG   \n",
       "579        Genes with probable DDR role                            UPF1   \n",
       "580         FA (Fanconi anemia pathway)                            USP1   \n",
       "581                                 TLS                            USP1   \n",
       "582        Genes with probable DDR role                           UVRAG   \n",
       "583  TCR (Transcription coupled repair)                           UVSSA   \n",
       "584        Genes with probable DDR role                             VCP   \n",
       "585         FA (Fanconi anemia pathway)                           WDR48   \n",
       "586                                 TLS                           WDR48   \n",
       "587                         G1-CC phase                            WEE1   \n",
       "588                         G2-CC phase                            WEE1   \n",
       "589                Telomere maintenance                          WRAP53   \n",
       "590       HR (Homologous Recombination)                             WRN   \n",
       "591              LONG PATCH-BER factors                             WRN   \n",
       "592  TCR (Transcription coupled repair)                            XAB2   \n",
       "593  TCR (Transcription coupled repair)                             XPA   \n",
       "594          GGR (Global genome repair)                             XPC   \n",
       "595  TCR (Transcription coupled repair)                           XRCC1   \n",
       "596                            Alt-NHEJ                           XRCC1   \n",
       "597             SHORT PATCH-BER factors                           XRCC1   \n",
       "598                                NHEJ                           XRCC2   \n",
       "599                                NHEJ                           XRCC3   \n",
       "600                                NHEJ                           XRCC4   \n",
       "601                                NHEJ                           XRCC5   \n",
       "602                                NHEJ                           XRCC6   \n",
       "603        Genes with probable DDR role                        XRCC6BP1   \n",
       "604        Genes with probable DDR role                            YBX1   \n",
       "605        Genes with probable DDR role                          ZRANB3   \n",
       "606        Genes with probable DDR role                          ZSWIM7   \n",
       "\n",
       "         EdgeType  \n",
       "0    Child-Parent  \n",
       "1    Child-Parent  \n",
       "2    Child-Parent  \n",
       "3    Child-Parent  \n",
       "4    Child-Parent  \n",
       "5    Child-Parent  \n",
       "6    Child-Parent  \n",
       "7    Child-Parent  \n",
       "8    Child-Parent  \n",
       "9    Child-Parent  \n",
       "10   Child-Parent  \n",
       "11   Child-Parent  \n",
       "12   Child-Parent  \n",
       "13   Child-Parent  \n",
       "14   Child-Parent  \n",
       "15   Child-Parent  \n",
       "16   Child-Parent  \n",
       "17   Child-Parent  \n",
       "18   Child-Parent  \n",
       "19   Child-Parent  \n",
       "20   Child-Parent  \n",
       "21   Child-Parent  \n",
       "22   Child-Parent  \n",
       "23   Child-Parent  \n",
       "24   Child-Parent  \n",
       "25   Child-Parent  \n",
       "26   Child-Parent  \n",
       "27   Child-Parent  \n",
       "28   Child-Parent  \n",
       "29   Child-Parent  \n",
       "..            ...  \n",
       "577     Gene-Term  \n",
       "578     Gene-Term  \n",
       "579     Gene-Term  \n",
       "580     Gene-Term  \n",
       "581     Gene-Term  \n",
       "582     Gene-Term  \n",
       "583     Gene-Term  \n",
       "584     Gene-Term  \n",
       "585     Gene-Term  \n",
       "586     Gene-Term  \n",
       "587     Gene-Term  \n",
       "588     Gene-Term  \n",
       "589     Gene-Term  \n",
       "590     Gene-Term  \n",
       "591     Gene-Term  \n",
       "592     Gene-Term  \n",
       "593     Gene-Term  \n",
       "594     Gene-Term  \n",
       "595     Gene-Term  \n",
       "596     Gene-Term  \n",
       "597     Gene-Term  \n",
       "598     Gene-Term  \n",
       "599     Gene-Term  \n",
       "600     Gene-Term  \n",
       "601     Gene-Term  \n",
       "602     Gene-Term  \n",
       "603     Gene-Term  \n",
       "604     Gene-Term  \n",
       "605     Gene-Term  \n",
       "606     Gene-Term  \n",
       "\n",
       "[607 rows x 3 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont0 = pd.read_csv('data/pearl2015.txt', sep='\\t', header=None)\n",
    "ont2 = Ontology.from_table(ont0)\n",
    "print(ont2)\n",
    "ont2.to_table('pearl2015.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ABL1',\n",
       " 'ACD',\n",
       " 'ACTL6A',\n",
       " 'ACTR5',\n",
       " 'ACTR8',\n",
       " 'ALKBH1',\n",
       " 'ALKBH2',\n",
       " 'ALKBH3',\n",
       " 'AMN1',\n",
       " 'ANKRD28',\n",
       " 'ANKRD44',\n",
       " 'ANKRD52',\n",
       " 'APEX1',\n",
       " 'APEX2',\n",
       " 'APLF',\n",
       " 'APTX',\n",
       " 'ARID1A',\n",
       " 'ARID1B',\n",
       " 'ARID2',\n",
       " 'ASF1A',\n",
       " 'ASF1B',\n",
       " 'ATF2',\n",
       " 'ATM',\n",
       " 'ATR',\n",
       " 'ATRIP',\n",
       " 'ATRX',\n",
       " 'ATXN3',\n",
       " 'BAP1',\n",
       " 'BARD1',\n",
       " 'BAZ1A',\n",
       " 'BAZ1B',\n",
       " 'BCAS2',\n",
       " 'BLM',\n",
       " 'BRCA1',\n",
       " 'BRCA2',\n",
       " 'BRCC3',\n",
       " 'BRD7',\n",
       " 'BRIP1',\n",
       " 'BTG2',\n",
       " 'CCDC98',\n",
       " 'CCNA1',\n",
       " 'CCNA2',\n",
       " 'CCNB1',\n",
       " 'CCNB2',\n",
       " 'CCNB3',\n",
       " 'CCND1',\n",
       " 'CCND2',\n",
       " 'CCND3',\n",
       " 'CCNE1',\n",
       " 'CCNH',\n",
       " 'CCNO',\n",
       " 'CDC25A',\n",
       " 'CDC25B',\n",
       " 'CDC5L',\n",
       " 'CDK2',\n",
       " 'CDK4',\n",
       " 'CDK7',\n",
       " 'CDKN1A',\n",
       " 'CDKN2A',\n",
       " 'CDKN2D',\n",
       " 'CETN2',\n",
       " 'CHAF1A',\n",
       " 'CHEK1',\n",
       " 'CHEK2',\n",
       " 'CHRAC1',\n",
       " 'CIB1',\n",
       " 'CLK2',\n",
       " 'CLSPN',\n",
       " 'CNOT7',\n",
       " 'COPS2',\n",
       " 'COPS3',\n",
       " 'COPS4',\n",
       " 'COPS5',\n",
       " 'COPS6',\n",
       " 'COPS7A',\n",
       " 'COPS7B',\n",
       " 'COPS8',\n",
       " 'CRY1',\n",
       " 'CRY2',\n",
       " 'CSNK1D',\n",
       " 'CSNK1E',\n",
       " 'CTC1',\n",
       " 'CUL3',\n",
       " 'CUL4A',\n",
       " 'CUL5',\n",
       " 'DAXX',\n",
       " 'DCLRE1A',\n",
       " 'DCLRE1B',\n",
       " 'DCLRE1C',\n",
       " 'DDB1',\n",
       " 'DDB2',\n",
       " 'DDX11',\n",
       " 'DKC1',\n",
       " 'DLGAP5',\n",
       " 'DMC1',\n",
       " 'DNA2',\n",
       " 'DNMT1',\n",
       " 'DNTT',\n",
       " 'DUT',\n",
       " 'EID3',\n",
       " 'EME1',\n",
       " 'EME2',\n",
       " 'ERCC1',\n",
       " 'ERCC2',\n",
       " 'ERCC3',\n",
       " 'ERCC4',\n",
       " 'ERCC5',\n",
       " 'ERCC6',\n",
       " 'ERCC8',\n",
       " 'EXO1',\n",
       " 'EYA3',\n",
       " 'FAAP100',\n",
       " 'FAAP24',\n",
       " 'FAN1',\n",
       " 'FANCA',\n",
       " 'FANCB',\n",
       " 'FANCC',\n",
       " 'FANCD2',\n",
       " 'FANCE',\n",
       " 'FANCG',\n",
       " 'FANCI',\n",
       " 'FANCL',\n",
       " 'FANCM',\n",
       " 'FEN1',\n",
       " 'GADD45A',\n",
       " 'GADD45G',\n",
       " 'GAR1',\n",
       " 'GEN1',\n",
       " 'GPS1',\n",
       " 'GTF2H1',\n",
       " 'GTF2H2',\n",
       " 'GTF2H3',\n",
       " 'GTF2H4',\n",
       " 'GTF2H5',\n",
       " 'HELQ',\n",
       " 'HES1',\n",
       " 'HFM1',\n",
       " 'HINFP',\n",
       " 'HLTF',\n",
       " 'HMGB1',\n",
       " 'HMGB2',\n",
       " 'HUS1',\n",
       " 'HUS1B',\n",
       " 'IGHMBP2',\n",
       " 'INO80',\n",
       " 'INO80C',\n",
       " 'INO80D',\n",
       " 'INO80E',\n",
       " 'IP6K3',\n",
       " 'KAT2A',\n",
       " 'KAT5',\n",
       " 'LIG1',\n",
       " 'LIG3',\n",
       " 'LIG4',\n",
       " 'MAD2L2',\n",
       " 'MBD4',\n",
       " 'MCRS1',\n",
       " 'MDC1',\n",
       " 'MDM2',\n",
       " 'MDM4',\n",
       " 'MEN1',\n",
       " 'MGMT',\n",
       " 'MLH1',\n",
       " 'MLH3',\n",
       " 'MMS19',\n",
       " 'MMS22L',\n",
       " 'MNAT1',\n",
       " 'MPG',\n",
       " 'MPLKIP',\n",
       " 'MRE11A',\n",
       " 'MSH2',\n",
       " 'MSH3',\n",
       " 'MSH4',\n",
       " 'MSH5',\n",
       " 'MSH6',\n",
       " 'MTOR',\n",
       " 'MUS81',\n",
       " 'MUTYH',\n",
       " 'MVP',\n",
       " 'MYBBP1A',\n",
       " 'NABP1',\n",
       " 'NABP2',\n",
       " 'NBN',\n",
       " 'NCAPD2',\n",
       " 'NCAPD3',\n",
       " 'NCAPG',\n",
       " 'NCAPH',\n",
       " 'NCAPH2',\n",
       " 'NEIL1',\n",
       " 'NEIL2',\n",
       " 'NEIL3',\n",
       " 'NFATC2IP',\n",
       " 'NFRKB',\n",
       " 'NHEJ1',\n",
       " 'NHP2',\n",
       " 'NOP10',\n",
       " 'NPM1',\n",
       " 'NSMCE1',\n",
       " 'NSMCE2',\n",
       " 'NSMCE4A',\n",
       " 'NTHL1',\n",
       " 'NUDT1',\n",
       " 'OBFC2B',\n",
       " 'OGG1',\n",
       " 'PALB2',\n",
       " 'PARG',\n",
       " 'PARP1',\n",
       " 'PARP3',\n",
       " 'PARP4',\n",
       " 'PARPBP',\n",
       " 'PAXIP1',\n",
       " 'PBRM1',\n",
       " 'PCNA',\n",
       " 'PDS5A',\n",
       " 'PDS5B',\n",
       " 'PER1',\n",
       " 'PER2',\n",
       " 'PER3',\n",
       " 'PIAS1',\n",
       " 'PIAS2',\n",
       " 'PIAS3',\n",
       " 'PIAS4',\n",
       " 'PIF1',\n",
       " 'PINX1',\n",
       " 'PLRG1',\n",
       " 'PMS1',\n",
       " 'PMS2',\n",
       " 'PMS2P5',\n",
       " 'PNKP',\n",
       " 'POLA1',\n",
       " 'POLB',\n",
       " 'POLD1',\n",
       " 'POLD2',\n",
       " 'POLD3',\n",
       " 'POLD4',\n",
       " 'POLE',\n",
       " 'POLE2',\n",
       " 'POLE3',\n",
       " 'POLE4',\n",
       " 'POLG',\n",
       " 'POLG2',\n",
       " 'POLH',\n",
       " 'POLI',\n",
       " 'POLK',\n",
       " 'POLL',\n",
       " 'POLM',\n",
       " 'POLN',\n",
       " 'POLQ',\n",
       " 'POLR2A',\n",
       " 'POLR2B',\n",
       " 'POLR2C',\n",
       " 'POLR2D',\n",
       " 'POLR2E',\n",
       " 'POLR2F',\n",
       " 'POLR2G',\n",
       " 'POLR2H',\n",
       " 'POLR2I',\n",
       " 'POLR2J',\n",
       " 'POLR2K',\n",
       " 'POLR2L',\n",
       " 'POT1',\n",
       " 'PPP4C',\n",
       " 'PPP4R1',\n",
       " 'PPP4R2',\n",
       " 'PPP4R4',\n",
       " 'PPP6C',\n",
       " 'PPP6R1',\n",
       " 'PPP6R2',\n",
       " 'PPP6R3',\n",
       " 'PRKCG',\n",
       " 'PRKDC',\n",
       " 'PRMT6',\n",
       " 'RAD1',\n",
       " 'RAD17',\n",
       " 'RAD18',\n",
       " 'RAD21',\n",
       " 'RAD23A',\n",
       " 'RAD23B',\n",
       " 'RAD50',\n",
       " 'RAD51',\n",
       " 'RAD51B',\n",
       " 'RAD51C',\n",
       " 'RAD51D',\n",
       " 'RAD52',\n",
       " 'RAD54B',\n",
       " 'RAD54L',\n",
       " 'RAD9A',\n",
       " 'RAD9B',\n",
       " 'RASSF7',\n",
       " 'RBBP8',\n",
       " 'RBX1',\n",
       " 'RDM1',\n",
       " 'RECQL',\n",
       " 'RECQL4',\n",
       " 'RECQL5',\n",
       " 'REV1',\n",
       " 'REV3L',\n",
       " 'RFC1',\n",
       " 'RFC2',\n",
       " 'RFC3',\n",
       " 'RFC4',\n",
       " 'RFC5',\n",
       " 'RIF1',\n",
       " 'RMI2',\n",
       " 'RNF168',\n",
       " 'RNF4',\n",
       " 'RNF8',\n",
       " 'RPA1',\n",
       " 'RPA2',\n",
       " 'RPA3',\n",
       " 'RPA4',\n",
       " 'RPAIN',\n",
       " 'RPS27L',\n",
       " 'RRM1',\n",
       " 'RRM2',\n",
       " 'RRM2B',\n",
       " 'RTEL1',\n",
       " 'RUVBL1',\n",
       " 'RUVBL2',\n",
       " 'SETMAR',\n",
       " 'SETX',\n",
       " 'SFR1',\n",
       " 'SHFM1',\n",
       " 'SHPRH',\n",
       " 'SIRT1',\n",
       " 'SLX1A',\n",
       " 'SLX1B',\n",
       " 'SLX4',\n",
       " 'SMARCA2',\n",
       " 'SMARCA4',\n",
       " 'SMARCA5',\n",
       " 'SMARCAD1',\n",
       " 'SMARCAL1',\n",
       " 'SMARCB1',\n",
       " 'SMARCC1',\n",
       " 'SMARCC2',\n",
       " 'SMARCD1',\n",
       " 'SMARCE1',\n",
       " 'SMC1A',\n",
       " 'SMC1B',\n",
       " 'SMC2',\n",
       " 'SMC3',\n",
       " 'SMC4',\n",
       " 'SMC5',\n",
       " 'SMC6',\n",
       " 'SMG6',\n",
       " 'SMUG1',\n",
       " 'SOD1',\n",
       " 'SPO11',\n",
       " 'STAG1',\n",
       " 'STAG2',\n",
       " 'SUMO1',\n",
       " 'SUMO2',\n",
       " 'SUMO3',\n",
       " 'SUMO4',\n",
       " 'SWI5',\n",
       " 'TADA3',\n",
       " 'TCEA1',\n",
       " 'TDG',\n",
       " 'TDP1',\n",
       " 'TELO2',\n",
       " 'TEN1',\n",
       " 'TEP1',\n",
       " 'TERF1',\n",
       " 'TERF2',\n",
       " 'TERF2IP',\n",
       " 'TERT',\n",
       " 'TFPT',\n",
       " 'TIMELESS',\n",
       " 'TINF2',\n",
       " 'TIPIN',\n",
       " 'TNKS',\n",
       " 'TNP1',\n",
       " 'TONSL',\n",
       " 'TOP1',\n",
       " 'TOP2A',\n",
       " 'TOP2B',\n",
       " 'TOP3A',\n",
       " 'TOP3B',\n",
       " 'TOPBP1',\n",
       " 'TP53',\n",
       " 'TP53BP1',\n",
       " 'TP73',\n",
       " 'TREX1',\n",
       " 'TREX2',\n",
       " 'TRIM28',\n",
       " 'TTI1',\n",
       " 'TTI2',\n",
       " 'TYMS',\n",
       " 'UBA1',\n",
       " 'UBA2',\n",
       " 'UBB',\n",
       " 'UBC',\n",
       " 'UBD',\n",
       " 'UBE2A',\n",
       " 'UBE2B',\n",
       " 'UBE2I',\n",
       " 'UBE2N',\n",
       " 'UBE2T',\n",
       " 'UBE2V1',\n",
       " 'UBE2V2',\n",
       " 'UIMC1',\n",
       " 'UNG',\n",
       " 'UPF1',\n",
       " 'USP1',\n",
       " 'UVRAG',\n",
       " 'UVSSA',\n",
       " 'VCP',\n",
       " 'WDR48',\n",
       " 'WEE1',\n",
       " 'WRAP53',\n",
       " 'WRN',\n",
       " 'XAB2',\n",
       " 'XPA',\n",
       " 'XPC',\n",
       " 'XRCC1',\n",
       " 'XRCC2',\n",
       " 'XRCC3',\n",
       " 'XRCC4',\n",
       " 'XRCC5',\n",
       " 'XRCC6',\n",
       " 'XRCC6BP1',\n",
       " 'YBX1',\n",
       " 'ZRANB3',\n",
       " 'ZSWIM7']"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matchers = ont2.genes\n",
    "matching = [s for s in matchers if any(xs in s for xs in go_human_symbol.genes)]\n",
    "matching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "425"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(matching)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "233"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matchers = ont.genes\n",
    "matching = [s for s in matchers if any(xs in s for xs in ont2.genes)]\n",
    "len(matching)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "[({218, 219, 220, 221, 222}, 0.5802251813002428), ({109, 111, 112, 113, 114}, 0.6067056040722213), ({192, 193, 170}, 0.6835762943989093), ({202, 127, 138}, 0.7005106393471607), ({248, 140, 247}, 0.7174214438597685), ({139, 141, 143}, 0.7297438599366777), ({232, 230, 231}, 0.7319602600860855), ({85, 86, 87}, 0.7359528143738937), ({142, 247}, 0.8489723014245765), ({264, 265}, 0.8565835752200133), ({177, 178}, 0.8597459057197241), ({168, 120}, 0.8650893417644118), ({165, 246}, 0.8719422626098627), ({50, 36}, 0.87737607502982), ({170, 171}, 0.8852184524534124), ({84, 85}, 0.8923500122737862), ({128, 132}, 0.9261346565729521), ({164, 246}, 0.9482938968281389), ({251, 252}, 0.9495998443046346), ({105, 300}, 0.9847271483003348)]\n",
    "[(1, -3339724.0), (3, -3340049.0), (0, -3340576.0), (10, -3341805.0), (4, -3342391.0), (7, -3342517.0), (14, -3344757.2), (8, -3344895.5), (6, -3344945.2), (2, -3344954.5), (12, -3345258.0), (11, -3346821.5), (5, -3346951.2), (13, -3347552.0), (15, -3348224.0), (9, -3360879.8)]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:.ddot]",
   "language": "python",
   "name": "conda-env-.ddot-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
